--- net/minecraft/src/AbstractTexture.java
+++ net/minecraft/src/AbstractTexture.java
@@ -1,13 +1,26 @@
 package net.minecraft.src;
 
+import org.lwjgl.opengl.GL11;
+
 public abstract class AbstractTexture implements TextureObject {
-	protected int glTextureId = -1;
+	public int glTextureId = -1;
 
 	public int getGlTextureId() {
-		if(this.glTextureId == -1) {
+		if (this.glTextureId == -1) {
 			this.glTextureId = TextureUtil.glGenTextures();
 		}
 
 		return this.glTextureId;
+	}
+
+	public void unloadGLTexture() {
+		if (this.glTextureId >= 0) {
+			GL11.glDeleteTextures(this.glTextureId);
+			this.glTextureId = -1;
+		}
+	}
+
+	public void finalize() {
+		this.unloadGLTexture();
 	}
 }
