--- net/minecraft/src/AchievementMap.java
+++ net/minecraft/src/AchievementMap.java
@@ -6,31 +6,34 @@
 import java.util.Map;
 
 public class AchievementMap {
+
+	/** Holds the singleton instance of AchievementMap. */
 	public static AchievementMap instance = new AchievementMap();
-	private Map b = new HashMap();
+
+	/** Maps a achievement id with it's unique GUID. */
+	private Map guidMap = new HashMap();
 
 	private AchievementMap() {
 		try {
 			BufferedReader var1 = new BufferedReader(new InputStreamReader(AchievementMap.class.getResourceAsStream("/achievement/map.txt")));
-
-			while(true) {
-				String var2 = var1.readLine();
-				if(var2 == null) {
-					var1.close();
-					break;
-				}
-
+			String var2;
+
+			while ((var2 = var1.readLine()) != null) {
 				String[] var3 = var2.split(",");
 				int var4 = Integer.parseInt(var3[0]);
-				this.b.put(Integer.valueOf(var4), var3[1]);
+				this.guidMap.put(Integer.valueOf(var4), var3[1]);
 			}
+
+			var1.close();
 		} catch (Exception var5) {
 			var5.printStackTrace();
 		}
-
 	}
 
-	public static String getGuid(int var0) {
-		return (String)instance.b.get(Integer.valueOf(var0));
+	/**
+	 * Returns the unique GUID of a achievement id.
+	 */
+	public static String getGuid(int par0) {
+		return (String)instance.guidMap.get(Integer.valueOf(par0));
 	}
 }
