--- net/minecraft/src/BlockDirectional.java
+++ net/minecraft/src/BlockDirectional.java
@@ -1,11 +1,14 @@
 package net.minecraft.src;
 
 public abstract class BlockDirectional extends Block {
-	protected BlockDirectional(int var1, Material var2) {
-		super(var1, var2);
+	protected BlockDirectional(int par1, Material par2Material) {
+		super(par1, par2Material);
 	}
 
-	public static int getDirection(int var0) {
-		return var0 & 3;
+	/**
+	 * Returns the orentation value from the specified metadata
+	 */
+	public static int getDirection(int par0) {
+		return par0 & 3;
 	}
 }
