--- net/minecraft/src/BlockSourceImpl.java
+++ net/minecraft/src/BlockSourceImpl.java
@@ -6,11 +6,11 @@
 	private final int yPos;
 	private final int zPos;
 
-	public BlockSourceImpl(World var1, int var2, int var3, int var4) {
-		this.worldObj = var1;
-		this.xPos = var2;
-		this.yPos = var3;
-		this.zPos = var4;
+	public BlockSourceImpl(World par1World, int par2, int par3, int par4) {
+		this.worldObj = par1World;
+		this.xPos = par2;
+		this.yPos = par3;
+		this.zPos = par4;
 	}
 
 	public World getWorld() {
