--- net/minecraft/src/BlockPressurePlate.java
+++ net/minecraft/src/BlockPressurePlate.java
@@ -4,41 +4,55 @@
 import java.util.List;
 
 public class BlockPressurePlate extends BlockBasePressurePlate {
+
+	/** The mob type that can trigger this pressure plate. */
 	private EnumMobType triggerMobType;
 
-	protected BlockPressurePlate(int var1, String var2, Material var3, EnumMobType var4) {
-		super(var1, var2, var3);
-		this.triggerMobType = var4;
-	}
-
-	protected int getMetaFromWeight(int var1) {
-		return var1 > 0 ? 1 : 0;
-	}
-
-	protected int getPowerSupply(int var1) {
-		return var1 == 1 ? 15 : 0;
-	}
-
-	protected int getPlateState(World var1, int var2, int var3, int var4) {
+	protected BlockPressurePlate(int par1, String par2Str, Material par3Material, EnumMobType par4EnumMobType) {
+		super(par1, par2Str, par3Material);
+		this.triggerMobType = par4EnumMobType;
+	}
+
+	/**
+	 * Argument is weight (0-15). Return the metadata to be set because of it.
+	 */
+	protected int getMetaFromWeight(int par1) {
+		return par1 > 0 ? 1 : 0;
+	}
+
+	/**
+	 * Argument is metadata. Returns power level (0-15)
+	 */
+	protected int getPowerSupply(int par1) {
+		return par1 == 1 ? 15 : 0;
+	}
+
+	/**
+	 * Returns the current state of the pressure plate. Returns a value between 0 and 15 based on the number of items on
+	 * it.
+	 */
+	protected int getPlateState(World par1World, int par2, int par3, int par4) {
 		List var5 = null;
-		if(this.triggerMobType == EnumMobType.everything) {
-			var5 = var1.getEntitiesWithinAABBExcludingEntity((Entity)null, this.getSensitiveAABB(var2, var3, var4));
-		}
-
-		if(this.triggerMobType == EnumMobType.mobs) {
-			var5 = var1.getEntitiesWithinAABB(EntityLivingBase.class, this.getSensitiveAABB(var2, var3, var4));
-		}
-
-		if(this.triggerMobType == EnumMobType.players) {
-			var5 = var1.getEntitiesWithinAABB(EntityPlayer.class, this.getSensitiveAABB(var2, var3, var4));
-		}
-
-		if(var5 != null && !var5.isEmpty()) {
+
+		if (this.triggerMobType == EnumMobType.everything) {
+			var5 = par1World.getEntitiesWithinAABBExcludingEntity((Entity)null, this.getSensitiveAABB(par2, par3, par4));
+		}
+
+		if (this.triggerMobType == EnumMobType.mobs) {
+			var5 = par1World.getEntitiesWithinAABB(EntityLivingBase.class, this.getSensitiveAABB(par2, par3, par4));
+		}
+
+		if (this.triggerMobType == EnumMobType.players) {
+			var5 = par1World.getEntitiesWithinAABB(EntityPlayer.class, this.getSensitiveAABB(par2, par3, par4));
+		}
+
+		if (var5 != null && !var5.isEmpty()) {
 			Iterator var6 = var5.iterator();
 
-			while(var6.hasNext()) {
+			while (var6.hasNext()) {
 				Entity var7 = (Entity)var6.next();
-				if(!var7.doesEntityNotTriggerPressurePlate()) {
+
+				if (!var7.doesEntityNotTriggerPressurePlate()) {
 					return 15;
 				}
 			}
