--- net/minecraft/src/BlockDeadBush.java
+++ net/minecraft/src/BlockDeadBush.java
@@ -3,27 +3,37 @@
 import java.util.Random;
 
 public class BlockDeadBush extends BlockFlower {
-	protected BlockDeadBush(int var1) {
-		super(var1, Material.vine);
+	protected BlockDeadBush(int par1) {
+		super(par1, Material.vine);
 		float var2 = 0.4F;
 		this.setBlockBounds(0.5F - var2, 0.0F, 0.5F - var2, 0.5F + var2, 0.8F, 0.5F + var2);
 	}
 
-	protected boolean canThisPlantGrowOnThisBlockID(int var1) {
-		return var1 == Block.sand.blockID;
+	/**
+	 * Gets passed in the blockID of the block below and supposed to return true if its allowed to grow on the type of
+	 * blockID passed in. Args: blockID
+	 */
+	protected boolean canThisPlantGrowOnThisBlockID(int par1) {
+		return par1 == Block.sand.blockID;
 	}
 
-	public int idDropped(int var1, Random var2, int var3) {
+	/**
+	 * Returns the ID of the items to drop on destruction.
+	 */
+	public int idDropped(int par1, Random par2Random, int par3) {
 		return -1;
 	}
 
-	public void harvestBlock(World var1, EntityPlayer var2, int var3, int var4, int var5, int var6) {
-		if(!var1.isRemote && var2.getCurrentEquippedItem() != null && var2.getCurrentEquippedItem().itemID == Item.shears.itemID) {
-			var2.addStat(StatList.mineBlockStatArray[this.blockID], 1);
-			this.dropBlockAsItem_do(var1, var3, var4, var5, new ItemStack(Block.deadBush, 1, var6));
+	/**
+	 * Called when the player destroys a block with an item that can harvest it. (i, j, k) are the coordinates of the block
+	 * and l is the block's subtype/damage.
+	 */
+	public void harvestBlock(World par1World, EntityPlayer par2EntityPlayer, int par3, int par4, int par5, int par6) {
+		if (!par1World.isRemote && par2EntityPlayer.getCurrentEquippedItem() != null && par2EntityPlayer.getCurrentEquippedItem().itemID == Item.shears.itemID) {
+			par2EntityPlayer.addStat(StatList.mineBlockStatArray[this.blockID], 1);
+			this.dropBlockAsItem_do(par1World, par3, par4, par5, new ItemStack(Block.deadBush, 1, par6));
 		} else {
-			super.harvestBlock(var1, var2, var3, var4, var5, var6);
+			super.harvestBlock(par1World, par2EntityPlayer, par3, par4, par5, par6);
 		}
-
 	}
 }
