--- net/minecraft/src/BlockContainer.java
+++ net/minecraft/src/BlockContainer.java
@@ -1,23 +1,34 @@
 package net.minecraft.src;
 
 public abstract class BlockContainer extends Block implements ITileEntityProvider {
-	protected BlockContainer(int var1, Material var2) {
-		super(var1, var2);
+	protected BlockContainer(int par1, Material par2Material) {
+		super(par1, par2Material);
 		this.isBlockContainer = true;
 	}
 
-	public void onBlockAdded(World var1, int var2, int var3, int var4) {
-		super.onBlockAdded(var1, var2, var3, var4);
-	}
-
-	public void breakBlock(World var1, int var2, int var3, int var4, int var5, int var6) {
-		super.breakBlock(var1, var2, var3, var4, var5, var6);
-		var1.removeBlockTileEntity(var2, var3, var4);
-	}
-
-	public boolean onBlockEventReceived(World var1, int var2, int var3, int var4, int var5, int var6) {
-		super.onBlockEventReceived(var1, var2, var3, var4, var5, var6);
-		TileEntity var7 = var1.getBlockTileEntity(var2, var3, var4);
-		return var7 != null ? var7.receiveClientEvent(var5, var6) : false;
+	/**
+	 * Called whenever the block is added into the world. Args: world, x, y, z
+	 */
+	public void onBlockAdded(World par1World, int par2, int par3, int par4) {
+		super.onBlockAdded(par1World, par2, par3, par4);
+	}
+
+	/**
+	 * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
+	 * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old metadata
+	 */
+	public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6) {
+		super.breakBlock(par1World, par2, par3, par4, par5, par6);
+		par1World.removeBlockTileEntity(par2, par3, par4);
+	}
+
+	/**
+	 * Called when the block receives a BlockEvent - see World.addBlockEvent. By default, passes it on to the tile entity
+	 * at this location. Args: world, x, y, z, blockID, EventID, event parameter
+	 */
+	public boolean onBlockEventReceived(World par1World, int par2, int par3, int par4, int par5, int par6) {
+		super.onBlockEventReceived(par1World, par2, par3, par4, par5, par6);
+		TileEntity var7 = par1World.getBlockTileEntity(par2, par3, par4);
+		return var7 != null ? var7.receiveClientEvent(par5, par6) : false;
 	}
 }
