--- net/minecraft/src/CallableTileEntityData.java
+++ net/minecraft/src/CallableTileEntityData.java
@@ -5,17 +5,18 @@
 class CallableTileEntityData implements Callable {
 	final TileEntity theTileEntity;
 
-	CallableTileEntityData(TileEntity var1) {
-		this.theTileEntity = var1;
+	CallableTileEntityData(TileEntity par1TileEntity) {
+		this.theTileEntity = par1TileEntity;
 	}
 
 	public String callTileEntityDataInfo() {
 		int var1 = this.theTileEntity.worldObj.getBlockMetadata(this.theTileEntity.xCoord, this.theTileEntity.yCoord, this.theTileEntity.zCoord);
-		if(var1 < 0) {
+
+		if (var1 < 0) {
 			return "Unknown? (Got " + var1 + ")";
 		} else {
-			String var2 = String.format("%4s", new Object[]{Integer.toBinaryString(var1)}).replace(" ", "0");
-			return String.format("%1$d / 0x%1$X / 0b%2$s", new Object[]{Integer.valueOf(var1), var2});
+			String var2 = String.format("%4s", new Object[] {Integer.toBinaryString(var1)}).replace(" ", "0");
+			return String.format("%1$d / 0x%1$X / 0b%2$s", new Object[] {Integer.valueOf(var1), var2});
 		}
 	}
 
