--- net/minecraft/src/CallableChunkPosHash.java
+++ net/minecraft/src/CallableChunkPosHash.java
@@ -4,13 +4,15 @@
 
 class CallableChunkPosHash implements Callable {
 	final int field_85165_a;
+
 	final int field_85163_b;
+
 	final MapGenStructure theMapStructureGenerator;
 
-	CallableChunkPosHash(MapGenStructure var1, int var2, int var3) {
-		this.theMapStructureGenerator = var1;
-		this.field_85165_a = var2;
-		this.field_85163_b = var3;
+	CallableChunkPosHash(MapGenStructure par1MapGenStructure, int par2, int par3) {
+		this.theMapStructureGenerator = par1MapGenStructure;
+		this.field_85165_a = par2;
+		this.field_85163_b = par3;
 	}
 
 	public String callChunkPositionHash() {
