--- net/minecraft/src/CallableJVMFlags.java
+++ net/minecraft/src/CallableJVMFlags.java
@@ -7,12 +7,17 @@
 import java.util.concurrent.Callable;
 
 class CallableJVMFlags implements Callable {
+
+	/** Reference to the CrashReport object. */
 	final CrashReport theCrashReport;
 
-	CallableJVMFlags(CrashReport var1) {
-		this.theCrashReport = var1;
+	CallableJVMFlags(CrashReport par1CrashReport) {
+		this.theCrashReport = par1CrashReport;
 	}
 
+	/**
+	 * Returns the number of JVM Flags along with the passed JVM Flags.
+	 */
 	public String getJVMFlagsAsString() {
 		RuntimeMXBean var1 = ManagementFactory.getRuntimeMXBean();
 		List var2 = var1.getInputArguments();
@@ -20,10 +25,11 @@
 		StringBuilder var4 = new StringBuilder();
 		Iterator var5 = var2.iterator();
 
-		while(var5.hasNext()) {
+		while (var5.hasNext()) {
 			String var6 = (String)var5.next();
-			if(var6.startsWith("-X")) {
-				if(var3++ > 0) {
+
+			if (var6.startsWith("-X")) {
+				if (var3++ > 0) {
 					var4.append(" ");
 				}
 
@@ -31,7 +37,7 @@
 			}
 		}
 
-		return String.format("%d total; %s", new Object[]{Integer.valueOf(var3), var4.toString()});
+		return String.format("%d total; %s", new Object[] {Integer.valueOf(var3), var4.toString()});
 	}
 
 	public Object call() {
