--- net/minecraft/src/CallablePacketID.java
+++ net/minecraft/src/CallablePacketID.java
@@ -4,11 +4,12 @@
 
 class CallablePacketID implements Callable {
 	final Packet thePacket;
+
 	final NetServerHandler theNetServerHandler;
 
-	CallablePacketID(NetServerHandler var1, Packet var2) {
-		this.theNetServerHandler = var1;
-		this.thePacket = var2;
+	CallablePacketID(NetServerHandler par1NetServerHandler, Packet par2Packet) {
+		this.theNetServerHandler = par1NetServerHandler;
+		this.thePacket = par2Packet;
 	}
 
 	public String callPacketID() {
