--- net/minecraft/src/CallablePacketClass.java
+++ net/minecraft/src/CallablePacketClass.java
@@ -4,11 +4,12 @@
 
 class CallablePacketClass implements Callable {
 	final Packet thePacket;
+
 	final NetServerHandler theNetServerHandler;
 
-	CallablePacketClass(NetServerHandler var1, Packet var2) {
-		this.theNetServerHandler = var1;
-		this.thePacket = var2;
+	CallablePacketClass(NetServerHandler par1NetServerHandler, Packet par2Packet) {
+		this.theNetServerHandler = par1NetServerHandler;
+		this.thePacket = par2Packet;
 	}
 
 	public String getPacketClass() {
