--- net/minecraft/src/CallableLvl3.java
+++ net/minecraft/src/CallableLvl3.java
@@ -3,12 +3,17 @@
 import java.util.concurrent.Callable;
 
 class CallableLvl3 implements Callable {
+
+	/** Reference to the World object. */
 	final World theWorld;
 
-	CallableLvl3(World var1) {
-		this.theWorld = var1;
+	CallableLvl3(World par1World) {
+		this.theWorld = par1World;
 	}
 
+	/**
+	 * Returns the result of the ChunkProvider's makeString
+	 */
 	public String getChunkProvider() {
 		return this.theWorld.chunkProvider.makeString();
 	}
