--- net/minecraft/src/CallableUpdatingScreenName.java
+++ net/minecraft/src/CallableUpdatingScreenName.java
@@ -5,8 +5,8 @@
 class CallableUpdatingScreenName implements Callable {
 	final Minecraft theMinecraft;
 
-	CallableUpdatingScreenName(Minecraft var1) {
-		this.theMinecraft = var1;
+	CallableUpdatingScreenName(Minecraft par1Minecraft) {
+		this.theMinecraft = par1Minecraft;
 	}
 
 	public String callUpdatingScreenName() {
