--- net/minecraft/src/ColorizerGrass.java
+++ net/minecraft/src/ColorizerGrass.java
@@ -1,16 +1,23 @@
 package net.minecraft.src;
 
 public class ColorizerGrass {
-	private static int[] grassBuffer = new int[65536];
-
-	public static void setGrassBiomeColorizer(int[] var0) {
-		grassBuffer = var0;
+
+	/** Color buffer for grass */
+	// MCPatcher Start - private -> public
+	public static int[] grassBuffer = new int[65536];
+	// MCPatcher End
+
+	public static void setGrassBiomeColorizer(int[] par0ArrayOfInteger) {
+		grassBuffer = par0ArrayOfInteger;
 	}
 
-	public static int getGrassColor(double var0, double var2) {
-		var2 *= var0;
-		int var4 = (int)((1.0D - var0) * 255.0D);
-		int var5 = (int)((1.0D - var2) * 255.0D);
+	/**
+	 * Gets grass color from temperature and humidity. Args: temperature, humidity
+	 */
+	public static int getGrassColor(double par0, double par2) {
+		par2 *= par0;
+		int var4 = (int)((1.0D - par0) * 255.0D);
+		int var5 = (int)((1.0D - par2) * 255.0D);
 		return grassBuffer[var5 << 8 | var4];
 	}
 }
