--- net/minecraft/src/CommandServerPardonIp.java
+++ net/minecraft/src/CommandServerPardonIp.java
@@ -9,24 +9,31 @@
 		return "pardon-ip";
 	}
 
+	/**
+	 * Return the required permission level for this command.
+	 */
 	public int getRequiredPermissionLevel() {
 		return 3;
 	}
 
-	public boolean canCommandSenderUseCommand(ICommandSender var1) {
-		return MinecraftServer.getServer().getConfigurationManager().getBannedIPs().isListActive() && super.canCommandSenderUseCommand(var1);
+	/**
+	 * Returns true if the given command sender is allowed to use this command.
+	 */
+	public boolean canCommandSenderUseCommand(ICommandSender par1ICommandSender) {
+		return MinecraftServer.getServer().getConfigurationManager().getBannedIPs().isListActive() && super.canCommandSenderUseCommand(par1ICommandSender);
 	}
 
-	public String getCommandUsage(ICommandSender var1) {
+	public String getCommandUsage(ICommandSender par1ICommandSender) {
 		return "commands.unbanip.usage";
 	}
 
-	public void processCommand(ICommandSender var1, String[] var2) {
-		if(var2.length == 1 && var2[0].length() > 1) {
-			Matcher var3 = CommandServerBanIp.a.matcher(var2[0]);
-			if(var3.matches()) {
-				MinecraftServer.getServer().getConfigurationManager().getBannedIPs().remove(var2[0]);
-				notifyAdmins(var1, "commands.unbanip.success", new Object[]{var2[0]});
+	public void processCommand(ICommandSender par1ICommandSender, String[] par2ArrayOfStr) {
+		if (par2ArrayOfStr.length == 1 && par2ArrayOfStr[0].length() > 1) {
+			Matcher var3 = CommandServerBanIp.IPv4Pattern.matcher(par2ArrayOfStr[0]);
+
+			if (var3.matches()) {
+				MinecraftServer.getServer().getConfigurationManager().getBannedIPs().remove(par2ArrayOfStr[0]);
+				notifyAdmins(par1ICommandSender, "commands.unbanip.success", new Object[] {par2ArrayOfStr[0]});
 			} else {
 				throw new SyntaxErrorException("commands.unbanip.invalid", new Object[0]);
 			}
@@ -35,7 +42,10 @@
 		}
 	}
 
-	public List addTabCompletionOptions(ICommandSender var1, String[] var2) {
-		return var2.length == 1 ? getListOfStringsFromIterableMatchingLastWord(var2, MinecraftServer.getServer().getConfigurationManager().getBannedIPs().getBannedList().keySet()) : null;
+	/**
+	 * Adds the strings available in this command to the given list of tab completion options.
+	 */
+	public List addTabCompletionOptions(ICommandSender par1ICommandSender, String[] par2ArrayOfStr) {
+		return par2ArrayOfStr.length == 1 ? getListOfStringsFromIterableMatchingLastWord(par2ArrayOfStr, MinecraftServer.getServer().getConfigurationManager().getBannedIPs().getBannedList().keySet()) : null;
 	}
 }
