--- net/minecraft/src/CommandToggleDownfall.java
+++ net/minecraft/src/CommandToggleDownfall.java
@@ -7,21 +7,27 @@
 		return "toggledownfall";
 	}
 
+	/**
+	 * Return the required permission level for this command.
+	 */
 	public int getRequiredPermissionLevel() {
 		return 2;
 	}
 
-	public String getCommandUsage(ICommandSender var1) {
+	public String getCommandUsage(ICommandSender par1ICommandSender) {
 		return "commands.downfall.usage";
 	}
 
-	public void processCommand(ICommandSender var1, String[] var2) {
+	public void processCommand(ICommandSender par1ICommandSender, String[] par2ArrayOfStr) {
 		this.toggleDownfall();
-		notifyAdmins(var1, "commands.downfall.success", new Object[0]);
+		notifyAdmins(par1ICommandSender, "commands.downfall.success", new Object[0]);
 	}
 
+	/**
+	 * Toggle rain and enable thundering.
+	 */
 	protected void toggleDownfall() {
-		MinecraftServer.getServer().worldServers[0].commandToggleDownfall();
+		MinecraftServer.getServer().worldServers[0].toggleRain();
 		MinecraftServer.getServer().worldServers[0].getWorldInfo().setThundering(true);
 	}
 }
