--- net/minecraft/src/ComponentStrongholdLeftTurn.java
+++ net/minecraft/src/ComponentStrongholdLeftTurn.java
@@ -4,40 +4,46 @@
 import java.util.Random;
 
 public class ComponentStrongholdLeftTurn extends ComponentStronghold {
-	public ComponentStrongholdLeftTurn() {
-	}
-
-	public ComponentStrongholdLeftTurn(int var1, Random var2, StructureBoundingBox var3, int var4) {
-		super(var1);
-		this.coordBaseMode = var4;
-		this.field_143013_d = this.getRandomDoor(var2);
-		this.boundingBox = var3;
-	}
-
-	public void buildComponent(StructureComponent var1, List var2, Random var3) {
-		if(this.coordBaseMode != 2 && this.coordBaseMode != 3) {
-			this.getNextComponentZ((ComponentStrongholdStairs2)var1, var2, var3, 1, 1);
+	public ComponentStrongholdLeftTurn() {}
+
+	public ComponentStrongholdLeftTurn(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4) {
+		super(par1);
+		this.coordBaseMode = par4;
+		this.field_143013_d = this.getRandomDoor(par2Random);
+		this.boundingBox = par3StructureBoundingBox;
+	}
+
+	/**
+	 * Initiates construction of the Structure Component picked, at the current Location of StructGen
+	 */
+	public void buildComponent(StructureComponent par1StructureComponent, List par2List, Random par3Random) {
+		if (this.coordBaseMode != 2 && this.coordBaseMode != 3) {
+			this.getNextComponentZ((ComponentStrongholdStairs2)par1StructureComponent, par2List, par3Random, 1, 1);
 		} else {
-			this.getNextComponentX((ComponentStrongholdStairs2)var1, var2, var3, 1, 1);
+			this.getNextComponentX((ComponentStrongholdStairs2)par1StructureComponent, par2List, par3Random, 1, 1);
 		}
-
-	}
-
-	public static ComponentStrongholdLeftTurn findValidPlacement(List var0, Random var1, int var2, int var3, int var4, int var5, int var6) {
-		StructureBoundingBox var7 = StructureBoundingBox.getComponentToAddBoundingBox(var2, var3, var4, -1, -1, 0, 5, 5, 5, var5);
-		return canStrongholdGoDeeper(var7) && StructureComponent.findIntersecting(var0, var7) == null ? new ComponentStrongholdLeftTurn(var6, var1, var7, var5) : null;
-	}
-
-	public boolean addComponentParts(World var1, Random var2, StructureBoundingBox var3) {
-		if(this.isLiquidInStructureBoundingBox(var1, var3)) {
+	}
+
+	public static ComponentStrongholdLeftTurn findValidPlacement(List par0List, Random par1Random, int par2, int par3, int par4, int par5, int par6) {
+		StructureBoundingBox var7 = StructureBoundingBox.getComponentToAddBoundingBox(par2, par3, par4, -1, -1, 0, 5, 5, 5, par5);
+		return canStrongholdGoDeeper(var7) && StructureComponent.findIntersecting(par0List, var7) == null ? new ComponentStrongholdLeftTurn(par6, par1Random, var7, par5) : null;
+	}
+
+	/**
+	 * second Part of Structure generating, this for example places Spiderwebs, Mob Spawners, it closes Mineshafts at the
+	 * end, it adds Fences...
+	 */
+	public boolean addComponentParts(World par1World, Random par2Random, StructureBoundingBox par3StructureBoundingBox) {
+		if (this.isLiquidInStructureBoundingBox(par1World, par3StructureBoundingBox)) {
 			return false;
 		} else {
-			this.fillWithRandomizedBlocks(var1, var3, 0, 0, 0, 4, 4, 4, true, var2, StructureStrongholdPieces.getStrongholdStones());
-			this.placeDoor(var1, var2, var3, this.field_143013_d, 1, 1, 0);
-			if(this.coordBaseMode != 2 && this.coordBaseMode != 3) {
-				this.fillWithBlocks(var1, var3, 4, 1, 1, 4, 3, 3, 0, 0, false);
+			this.fillWithRandomizedBlocks(par1World, par3StructureBoundingBox, 0, 0, 0, 4, 4, 4, true, par2Random, StructureStrongholdPieces.getStrongholdStones());
+			this.placeDoor(par1World, par2Random, par3StructureBoundingBox, this.field_143013_d, 1, 1, 0);
+
+			if (this.coordBaseMode != 2 && this.coordBaseMode != 3) {
+				this.fillWithBlocks(par1World, par3StructureBoundingBox, 4, 1, 1, 4, 3, 3, 0, 0, false);
 			} else {
-				this.fillWithBlocks(var1, var3, 0, 1, 1, 0, 3, 3, 0, 0, false);
+				this.fillWithBlocks(par1World, par3StructureBoundingBox, 0, 1, 1, 0, 3, 3, 0, 0, false);
 			}
 
 			return true;
