--- net/minecraft/src/ComparatorClassSorter.java
+++ net/minecraft/src/ComparatorClassSorter.java
@@ -5,17 +5,17 @@
 class ComparatorClassSorter implements Comparator {
 	final CallableSuspiciousClasses theSuspiciousClasses;
 
-	ComparatorClassSorter(CallableSuspiciousClasses var1) {
-		this.theSuspiciousClasses = var1;
+	ComparatorClassSorter(CallableSuspiciousClasses par1CallableSuspiciousClasses) {
+		this.theSuspiciousClasses = par1CallableSuspiciousClasses;
 	}
 
-	public int func_85081_a(Class var1, Class var2) {
-		String var3 = var1.getPackage() == null ? "" : var1.getPackage().getName();
-		String var4 = var2.getPackage() == null ? "" : var2.getPackage().getName();
+	public int func_85081_a(Class par1Class, Class par2Class) {
+		String var3 = par1Class.getPackage() == null ? "" : par1Class.getPackage().getName();
+		String var4 = par2Class.getPackage() == null ? "" : par2Class.getPackage().getName();
 		return var3.compareTo(var4);
 	}
 
-	public int compare(Object var1, Object var2) {
-		return this.func_85081_a((Class)var1, (Class)var2);
+	public int compare(Object par1Obj, Object par2Obj) {
+		return this.func_85081_a((Class)par1Obj, (Class)par2Obj);
 	}
 }
