--- net/minecraft/src/CreativeTabRedstone.java
+++ net/minecraft/src/CreativeTabRedstone.java
@@ -1,10 +1,13 @@
 package net.minecraft.src;
 
 final class CreativeTabRedstone extends CreativeTabs {
-	CreativeTabRedstone(int var1, String var2) {
-		super(var1, var2);
+	CreativeTabRedstone(int par1, String par2Str) {
+		super(par1, par2Str);
 	}
 
+	/**
+	 * the itemID for the item to be displayed on the tab
+	 */
 	public int getTabIconItemIndex() {
 		return Block.railPowered.blockID;
 	}
