--- net/minecraft/src/DedicatedServerCommandThread.java
+++ net/minecraft/src/DedicatedServerCommandThread.java
@@ -7,25 +7,20 @@
 class DedicatedServerCommandThread extends Thread {
 	final DedicatedServer server;
 
-	DedicatedServerCommandThread(DedicatedServer var1) {
-		this.server = var1;
+	DedicatedServerCommandThread(DedicatedServer par1DedicatedServer) {
+		this.server = par1DedicatedServer;
 	}
 
 	public void run() {
 		BufferedReader var1 = new BufferedReader(new InputStreamReader(System.in));
+		String var2;
 
 		try {
-			while(!this.server.isServerStopped() && this.server.isServerRunning()) {
-				String var2 = var1.readLine();
-				if(var2 == null) {
-					break;
-				}
-
+			while (!this.server.isServerStopped() && this.server.isServerRunning() && (var2 = var1.readLine()) != null) {
 				this.server.addPendingCommand(var2, this.server);
 			}
 		} catch (IOException var4) {
 			var4.printStackTrace();
 		}
-
 	}
 }
