--- net/minecraft/src/DispenserBehaviorArrow.java
+++ net/minecraft/src/DispenserBehaviorArrow.java
@@ -1,8 +1,12 @@
 package net.minecraft.src;
 
 final class DispenserBehaviorArrow extends BehaviorProjectileDispense {
-	protected IProjectile getProjectileEntity(World var1, IPosition var2) {
-		EntityArrow var3 = new EntityArrow(var1, var2.getX(), var2.getY(), var2.getZ());
+
+	/**
+	 * Return the projectile entity spawned by this dispense behavior.
+	 */
+	protected IProjectile getProjectileEntity(World par1World, IPosition par2IPosition) {
+		EntityArrow var3 = new EntityArrow(par1World, par2IPosition.getX(), par2IPosition.getY(), par2IPosition.getZ());
 		var3.canBePickedUp = 1;
 		return var3;
 	}
