--- net/minecraft/src/EntityFootStepFX.java
+++ net/minecraft/src/EntityFootStepFX.java
@@ -8,24 +8,25 @@
 	private int footstepMaxAge;
 	private TextureManager currentFootSteps;
 
-	public EntityFootStepFX(TextureManager var1, World var2, double var3, double var5, double var7) {
-		super(var2, var3, var5, var7, 0.0D, 0.0D, 0.0D);
-		this.currentFootSteps = var1;
+	public EntityFootStepFX(TextureManager par1TextureManager, World par2World, double par3, double par5, double par7) {
+		super(par2World, par3, par5, par7, 0.0D, 0.0D, 0.0D);
+		this.currentFootSteps = par1TextureManager;
 		this.motionX = this.motionY = this.motionZ = 0.0D;
 		this.footstepMaxAge = 200;
 	}
 
-	public void renderParticle(Tessellator var1, float var2, float var3, float var4, float var5, float var6, float var7) {
-		float var8 = ((float)this.footstepAge + var2) / (float)this.footstepMaxAge;
+	public void renderParticle(Tessellator par1Tessellator, float par2, float par3, float par4, float par5, float par6, float par7) {
+		float var8 = ((float)this.footstepAge + par2) / (float)this.footstepMaxAge;
 		var8 *= var8;
 		float var9 = 2.0F - var8 * 2.0F;
-		if(var9 > 1.0F) {
+
+		if (var9 > 1.0F) {
 			var9 = 1.0F;
 		}
 
 		var9 *= 0.2F;
 		GL11.glDisable(GL11.GL_LIGHTING);
-		float var10 = 2.0F / 16.0F;
+		float var10 = 0.125F;
 		float var11 = (float)(this.posX - interpPosX);
 		float var12 = (float)(this.posY - interpPosY);
 		float var13 = (float)(this.posZ - interpPosZ);
@@ -33,23 +34,26 @@
 		this.currentFootSteps.bindTexture(field_110126_a);
 		GL11.glEnable(GL11.GL_BLEND);
 		GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
-		var1.startDrawingQuads();
-		var1.setColorRGBA_F(var14, var14, var14, var9);
-		var1.addVertexWithUV((double)(var11 - var10), (double)var12, (double)(var13 + var10), 0.0D, 1.0D);
-		var1.addVertexWithUV((double)(var11 + var10), (double)var12, (double)(var13 + var10), 1.0D, 1.0D);
-		var1.addVertexWithUV((double)(var11 + var10), (double)var12, (double)(var13 - var10), 1.0D, 0.0D);
-		var1.addVertexWithUV((double)(var11 - var10), (double)var12, (double)(var13 - var10), 0.0D, 0.0D);
-		var1.draw();
+		par1Tessellator.startDrawingQuads();
+		par1Tessellator.setColorRGBA_F(var14, var14, var14, var9);
+		par1Tessellator.addVertexWithUV((double)(var11 - var10), (double)var12, (double)(var13 + var10), 0.0D, 1.0D);
+		par1Tessellator.addVertexWithUV((double)(var11 + var10), (double)var12, (double)(var13 + var10), 1.0D, 1.0D);
+		par1Tessellator.addVertexWithUV((double)(var11 + var10), (double)var12, (double)(var13 - var10), 1.0D, 0.0D);
+		par1Tessellator.addVertexWithUV((double)(var11 - var10), (double)var12, (double)(var13 - var10), 0.0D, 0.0D);
+		par1Tessellator.draw();
 		GL11.glDisable(GL11.GL_BLEND);
 		GL11.glEnable(GL11.GL_LIGHTING);
 	}
 
+	/**
+	 * Called to update the entity's position/logic.
+	 */
 	public void onUpdate() {
 		++this.footstepAge;
-		if(this.footstepAge == this.footstepMaxAge) {
+
+		if (this.footstepAge == this.footstepMaxAge) {
 			this.setDead();
 		}
-
 	}
 
 	public int getFXLayer() {
