--- net/minecraft/src/EntityHorseGroupData.java
+++ net/minecraft/src/EntityHorseGroupData.java
@@ -4,8 +4,8 @@
 	public int field_111107_a;
 	public int field_111106_b;
 
-	public EntityHorseGroupData(int var1, int var2) {
-		this.field_111107_a = var1;
-		this.field_111106_b = var2;
+	public EntityHorseGroupData(int par1, int par2) {
+		this.field_111107_a = par1;
+		this.field_111106_b = par2;
 	}
 }
