--- net/minecraft/src/EntityAuraFX.java
+++ net/minecraft/src/EntityAuraFX.java
@@ -1,22 +1,30 @@
 package net.minecraft.src;
 
+// MCPatcher Start
+import com.prupe.mcpatcher.cc.ColorizeEntity;
+import com.prupe.mcpatcher.cc.Colorizer;
+// MCPatcher End
+
 public class EntityAuraFX extends EntityFX {
-	public EntityAuraFX(World var1, double var2, double var4, double var6, double var8, double var10, double var12) {
-		super(var1, var2, var4, var6, var8, var10, var12);
-		float var14 = this.ab.nextFloat() * 0.1F + 0.2F;
+	public EntityAuraFX(World par1World, double par2, double par4, double par6, double par8, double par10, double par12) {
+		super(par1World, par2, par4, par6, par8, par10, par12);
+		float var14 = this.rand.nextFloat() * 0.1F + 0.2F;
 		this.particleRed = var14;
 		this.particleGreen = var14;
 		this.particleBlue = var14;
 		this.setParticleTextureIndex(0);
 		this.setSize(0.02F, 0.02F);
-		this.particleScale *= this.ab.nextFloat() * 0.6F + 0.5F;
-		this.motionX *= (double)0.02F;
-		this.motionY *= (double)0.02F;
-		this.motionZ *= (double)0.02F;
+		this.particleScale *= this.rand.nextFloat() * 0.6F + 0.5F;
+		this.motionX *= 0.019999999552965164D;
+		this.motionY *= 0.019999999552965164D;
+		this.motionZ *= 0.019999999552965164D;
 		this.particleMaxAge = (int)(20.0D / (Math.random() * 0.8D + 0.2D));
 		this.noClip = true;
 	}
 
+	/**
+	 * Called to update the entity's position/logic.
+	 */
 	public void onUpdate() {
 		this.prevPosX = this.posX;
 		this.prevPosY = this.posY;
@@ -25,9 +33,21 @@
 		this.motionX *= 0.99D;
 		this.motionY *= 0.99D;
 		this.motionZ *= 0.99D;
-		if(this.particleMaxAge-- <= 0) {
+
+		if (this.particleMaxAge-- <= 0) {
 			this.setDead();
 		}
-
-	}
+	}
+
+	// MCPatcher Start
+	public EntityAuraFX colorize() {
+		if (ColorizeEntity.computeMyceliumParticleColor()) {
+			this.particleRed = Colorizer.setColor[0];
+			this.particleGreen = Colorizer.setColor[1];
+			this.particleBlue = Colorizer.setColor[2];
+		}
+
+		return this;
+	}
+	// MCPatcher End
 }
