--- net/minecraft/src/EntityAIWander.java
+++ net/minecraft/src/EntityAIWander.java
@@ -7,20 +7,24 @@
 	private double zPosition;
 	private double speed;
 
-	public EntityAIWander(EntityCreature var1, double var2) {
-		this.entity = var1;
-		this.speed = var2;
+	public EntityAIWander(EntityCreature par1EntityCreature, double par2) {
+		this.entity = par1EntityCreature;
+		this.speed = par2;
 		this.setMutexBits(1);
 	}
 
+	/**
+	 * Returns whether the EntityAIBase should begin execution.
+	 */
 	public boolean shouldExecute() {
-		if(this.entity.getAge() >= 100) {
+		if (this.entity.getAge() >= 100) {
 			return false;
-		} else if(this.entity.getRNG().nextInt(120) != 0) {
+		} else if (this.entity.getRNG().nextInt(120) != 0) {
 			return false;
 		} else {
 			Vec3 var1 = RandomPositionGenerator.findRandomTarget(this.entity, 10, 7);
-			if(var1 == null) {
+
+			if (var1 == null) {
 				return false;
 			} else {
 				this.xPosition = var1.xCoord;
@@ -31,10 +35,16 @@
 		}
 	}
 
+	/**
+	 * Returns whether an in-progress EntityAIBase should continue executing
+	 */
 	public boolean continueExecuting() {
 		return !this.entity.getNavigator().noPath();
 	}
 
+	/**
+	 * Execute a one shot task or start executing a continuous task
+	 */
 	public void startExecuting() {
 		this.entity.getNavigator().tryMoveToXYZ(this.xPosition, this.yPosition, this.zPosition, this.speed);
 	}
