--- net/minecraft/src/EntityExplodeFX.java
+++ net/minecraft/src/EntityExplodeFX.java
@@ -1,34 +1,38 @@
 package net.minecraft.src;
 
 public class EntityExplodeFX extends EntityFX {
-	public EntityExplodeFX(World var1, double var2, double var4, double var6, double var8, double var10, double var12) {
-		super(var1, var2, var4, var6, var8, var10, var12);
-		this.motionX = var8 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-		this.motionY = var10 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-		this.motionZ = var12 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-		this.particleRed = this.particleGreen = this.particleBlue = this.ab.nextFloat() * 0.3F + 0.7F;
-		this.particleScale = this.ab.nextFloat() * this.ab.nextFloat() * 6.0F + 1.0F;
-		this.particleMaxAge = (int)(16.0D / ((double)this.ab.nextFloat() * 0.8D + 0.2D)) + 2;
+	public EntityExplodeFX(World par1World, double par2, double par4, double par6, double par8, double par10, double par12) {
+		super(par1World, par2, par4, par6, par8, par10, par12);
+		this.motionX = par8 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
+		this.motionY = par10 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
+		this.motionZ = par12 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
+		this.particleRed = this.particleGreen = this.particleBlue = this.rand.nextFloat() * 0.3F + 0.7F;
+		this.particleScale = this.rand.nextFloat() * this.rand.nextFloat() * 6.0F + 1.0F;
+		this.particleMaxAge = (int)(16.0D / ((double)this.rand.nextFloat() * 0.8D + 0.2D)) + 2;
 	}
 
+	/**
+	 * Called to update the entity's position/logic.
+	 */
 	public void onUpdate() {
 		this.prevPosX = this.posX;
 		this.prevPosY = this.posY;
 		this.prevPosZ = this.posZ;
-		if(this.particleAge++ >= this.particleMaxAge) {
+
+		if (this.particleAge++ >= this.particleMaxAge) {
 			this.setDead();
 		}
 
 		this.setParticleTextureIndex(7 - this.particleAge * 8 / this.particleMaxAge);
 		this.motionY += 0.004D;
 		this.moveEntity(this.motionX, this.motionY, this.motionZ);
-		this.motionX *= (double)0.9F;
-		this.motionY *= (double)0.9F;
-		this.motionZ *= (double)0.9F;
-		if(this.onGround) {
-			this.motionX *= (double)0.7F;
-			this.motionZ *= (double)0.7F;
-		}
+		this.motionX *= 0.8999999761581421D;
+		this.motionY *= 0.8999999761581421D;
+		this.motionZ *= 0.8999999761581421D;
 
+		if (this.onGround) {
+			this.motionX *= 0.699999988079071D;
+			this.motionZ *= 0.699999988079071D;
+		}
 	}
 }
