--- net/minecraft/src/EnumRarity.java
+++ net/minecraft/src/EnumRarity.java
@@ -6,11 +6,17 @@
 	rare(11, "Rare"),
 	epic(13, "Epic");
 
+	/**
+	 * A decimal representation of the hex color codes of a the color assigned to this rarity type. (13 becomes d as in
+	 * \247d which is light purple)
+	 */
 	public final int rarityColor;
-	public final String f;
-
-	private EnumRarity(int var3, String var4) {
-		this.rarityColor = var3;
-		this.f = var4;
+
+	/** Rarity name. */
+	public final String rarityName;
+
+	private EnumRarity(int par3, String par4Str) {
+		this.rarityColor = par3;
+		this.rarityName = par4Str;
 	}
 }
