--- net/minecraft/src/Frustrum.java
+++ net/minecraft/src/Frustrum.java
@@ -6,17 +6,23 @@
 	private double yPosition;
 	private double zPosition;
 
-	public void setPosition(double var1, double var3, double var5) {
-		this.xPosition = var1;
-		this.yPosition = var3;
-		this.zPosition = var5;
-	}
-
-	public boolean isBoxInFrustum(double var1, double var3, double var5, double var7, double var9, double var11) {
-		return this.clippingHelper.isBoxInFrustum(var1 - this.xPosition, var3 - this.yPosition, var5 - this.zPosition, var7 - this.xPosition, var9 - this.yPosition, var11 - this.zPosition);
-	}
-
-	public boolean isBoundingBoxInFrustum(AxisAlignedBB var1) {
-		return this.isBoxInFrustum(var1.minX, var1.minY, var1.minZ, var1.maxX, var1.maxY, var1.maxZ);
+	public void setPosition(double par1, double par3, double par5) {
+		this.xPosition = par1;
+		this.yPosition = par3;
+		this.zPosition = par5;
+	}
+
+	/**
+	 * Calls the clipping helper. Returns true if the box is inside all 6 clipping planes, otherwise returns false.
+	 */
+	public boolean isBoxInFrustum(double par1, double par3, double par5, double par7, double par9, double par11) {
+		return this.clippingHelper.isBoxInFrustum(par1 - this.xPosition, par3 - this.yPosition, par5 - this.zPosition, par7 - this.xPosition, par9 - this.yPosition, par11 - this.zPosition);
+	}
+
+	/**
+	 * Returns true if the bounding box is inside all 6 clipping planes, otherwise returns false.
+	 */
+	public boolean isBoundingBoxInFrustum(AxisAlignedBB par1AxisAlignedBB) {
+		return this.isBoxInFrustum(par1AxisAlignedBB.minX, par1AxisAlignedBB.minY, par1AxisAlignedBB.minZ, par1AxisAlignedBB.maxX, par1AxisAlignedBB.maxY, par1AxisAlignedBB.maxZ);
 	}
 }
