--- net/minecraft/src/GrassColorReloadListener.java
+++ net/minecraft/src/GrassColorReloadListener.java
@@ -5,11 +5,11 @@
 public class GrassColorReloadListener implements ResourceManagerReloadListener {
 	private static final ResourceLocation field_130078_a = new ResourceLocation("textures/colormap/grass.png");
 
-	public void onResourceManagerReload(ResourceManager var1) {
+	public void onResourceManagerReload(ResourceManager par1ResourceManager) {
 		try {
-			ColorizerGrass.setGrassBiomeColorizer(TextureUtil.readImageData(var1, field_130078_a));
+			ColorizerGrass.setGrassBiomeColorizer(TextureUtil.readImageData(par1ResourceManager, field_130078_a));
 		} catch (IOException var3) {
+			;
 		}
-
 	}
 }
