--- net/minecraft/src/GuiBeaconButtonConfirm.java
+++ net/minecraft/src/GuiBeaconButtonConfirm.java
@@ -1,14 +1,16 @@
 package net.minecraft.src;
 
 class GuiBeaconButtonConfirm extends GuiBeaconButton {
+
+	/** Beacon GUI this button belongs to. */
 	final GuiBeacon beaconGui;
 
-	public GuiBeaconButtonConfirm(GuiBeacon var1, int var2, int var3, int var4) {
-		super(var2, var3, var4, GuiBeacon.getBeaconGuiTextures(), 90, 220);
-		this.beaconGui = var1;
+	public GuiBeaconButtonConfirm(GuiBeacon par1GuiBeacon, int par2, int par3, int par4) {
+		super(par2, par3, par4, GuiBeacon.getBeaconGuiTextures(), 90, 220);
+		this.beaconGui = par1GuiBeacon;
 	}
 
-	public void func_82251_b(int var1, int var2) {
-		this.beaconGui.drawCreativeTabHoveringText(I18n.getString("gui.done"), var1, var2);
+	public void func_82251_b(int par1, int par2) {
+		this.beaconGui.drawCreativeTabHoveringText(I18n.getString("gui.done"), par1, par2);
 	}
 }
