--- net/minecraft/src/GuiButtonNextPage.java
+++ net/minecraft/src/GuiButtonNextPage.java
@@ -3,25 +3,33 @@
 import org.lwjgl.opengl.GL11;
 
 class GuiButtonNextPage extends GuiButton {
+
+	/**
+	 * True for pointing right (next page), false for pointing left (previous page).
+	 */
 	private final boolean nextPage;
 
-	public GuiButtonNextPage(int var1, int var2, int var3, boolean var4) {
-		super(var1, var2, var3, 23, 13, "");
-		this.nextPage = var4;
+	public GuiButtonNextPage(int par1, int par2, int par3, boolean par4) {
+		super(par1, par2, par3, 23, 13, "");
+		this.nextPage = par4;
 	}
 
-	public void drawButton(Minecraft var1, int var2, int var3) {
-		if(this.drawButton) {
-			boolean var4 = var2 >= this.xPosition && var3 >= this.yPosition && var2 < this.xPosition + this.width && var3 < this.yPosition + this.height;
+	/**
+	 * Draws this button to the screen.
+	 */
+	public void drawButton(Minecraft par1Minecraft, int par2, int par3) {
+		if (this.drawButton) {
+			boolean var4 = par2 >= this.xPosition && par3 >= this.yPosition && par2 < this.xPosition + this.width && par3 < this.yPosition + this.height;
 			GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-			var1.getTextureManager().bindTexture(GuiScreenBook.func_110404_g());
+			par1Minecraft.getTextureManager().bindTexture(GuiScreenBook.func_110404_g());
 			int var5 = 0;
 			int var6 = 192;
-			if(var4) {
+
+			if (var4) {
 				var5 += 23;
 			}
 
-			if(!this.nextPage) {
+			if (!this.nextPage) {
 				var6 += 13;
 			}
 
