--- net/minecraft/src/GuiDispenser.java
+++ net/minecraft/src/GuiDispenser.java
@@ -6,18 +6,24 @@
 	private static final ResourceLocation dispenserGuiTextures = new ResourceLocation("textures/gui/container/dispenser.png");
 	public TileEntityDispenser theDispenser;
 
-	public GuiDispenser(InventoryPlayer var1, TileEntityDispenser var2) {
-		super(new ContainerDispenser(var1, var2));
-		this.theDispenser = var2;
+	public GuiDispenser(InventoryPlayer par1InventoryPlayer, TileEntityDispenser par2TileEntityDispenser) {
+		super(new ContainerDispenser(par1InventoryPlayer, par2TileEntityDispenser));
+		this.theDispenser = par2TileEntityDispenser;
 	}
 
-	protected void drawGuiContainerForegroundLayer(int var1, int var2) {
+	/**
+	 * Draw the foreground layer for the GuiContainer (everything in front of the items)
+	 */
+	protected void drawGuiContainerForegroundLayer(int par1, int par2) {
 		String var3 = this.theDispenser.isInvNameLocalized() ? this.theDispenser.getInvName() : I18n.getString(this.theDispenser.getInvName());
 		this.fontRenderer.drawString(var3, this.xSize / 2 - this.fontRenderer.getStringWidth(var3) / 2, 6, 4210752);
 		this.fontRenderer.drawString(I18n.getString("container.inventory"), 8, this.ySize - 96 + 2, 4210752);
 	}
 
-	protected void drawGuiContainerBackgroundLayer(float var1, int var2, int var3) {
+	/**
+	 * Draw the background layer for the GuiContainer (everything behind the items)
+	 */
+	protected void drawGuiContainerBackgroundLayer(float par1, int par2, int par3) {
 		GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
 		this.mc.getTextureManager().bindTexture(dispenserGuiTextures);
 		int var4 = (this.width - this.xSize) / 2;
