--- net/minecraft/src/GuiSlotStatsGeneral.java
+++ net/minecraft/src/GuiSlotStatsGeneral.java
@@ -3,23 +3,34 @@
 class GuiSlotStatsGeneral extends GuiSlot {
 	final GuiStats statsGui;
 
-	public GuiSlotStatsGeneral(GuiStats var1) {
-		super(GuiStats.getMinecraft(var1), var1.width, var1.height, 32, var1.height - 64, 10);
-		this.statsGui = var1;
+	public GuiSlotStatsGeneral(GuiStats par1GuiStats) {
+		super(GuiStats.getMinecraft(par1GuiStats), par1GuiStats.width, par1GuiStats.height, 32, par1GuiStats.height - 64, 10);
+		this.statsGui = par1GuiStats;
 		this.setShowSelectionBox(false);
 	}
 
+	/**
+	 * Gets the size of the current slot list.
+	 */
 	protected int getSize() {
-		return StatList.c.size();
-	}
-
-	protected void elementClicked(int var1, boolean var2) {
-	}
-
-	protected boolean isSelected(int var1) {
+		return StatList.generalStats.size();
+	}
+
+	/**
+	 * the element in the slot that was clicked, boolean for wether it was double clicked or not
+	 */
+	protected void elementClicked(int par1, boolean par2) {}
+
+	/**
+	 * returns true if the element passed in is currently selected
+	 */
+	protected boolean isSelected(int par1) {
 		return false;
 	}
 
+	/**
+	 * return the height of the content being scrolled
+	 */
 	protected int getContentHeight() {
 		return this.getSize() * 10;
 	}
@@ -28,10 +39,10 @@
 		this.statsGui.drawDefaultBackground();
 	}
 
-	protected void drawSlot(int var1, int var2, int var3, int var4, Tessellator var5) {
-		StatBase var6 = (StatBase)StatList.c.get(var1);
-		this.statsGui.drawString(GuiStats.getFontRenderer1(this.statsGui), I18n.getString(var6.getName()), var2 + 2, var3 + 1, var1 % 2 == 0 ? 16777215 : 9474192);
+	protected void drawSlot(int par1, int par2, int par3, int par4, Tessellator par5Tessellator) {
+		StatBase var6 = (StatBase)StatList.generalStats.get(par1);
+		this.statsGui.drawString(GuiStats.getFontRenderer1(this.statsGui), I18n.getString(var6.getName()), par2 + 2, par3 + 1, par1 % 2 == 0 ? 16777215 : 9474192);
 		String var7 = var6.func_75968_a(GuiStats.getStatsFileWriter(this.statsGui).writeStat(var6));
-		this.statsGui.drawString(GuiStats.getFontRenderer2(this.statsGui), var7, var2 + 2 + 213 - GuiStats.getFontRenderer3(this.statsGui).getStringWidth(var7), var3 + 1, var1 % 2 == 0 ? 16777215 : 9474192);
+		this.statsGui.drawString(GuiStats.getFontRenderer2(this.statsGui), var7, par2 + 2 + 213 - GuiStats.getFontRenderer3(this.statsGui).getStringWidth(var7), par3 + 1, par1 % 2 == 0 ? 16777215 : 9474192);
 	}
 }
