--- net/minecraft/src/GuiSlotLanguage.java
+++ net/minecraft/src/GuiSlotLanguage.java
@@ -7,44 +7,56 @@
 import java.util.Map;
 
 class GuiSlotLanguage extends GuiSlot {
-	private final List b;
-	private final Map h;
+	private final List field_77251_g;
+	private final Map field_77253_h;
+
 	final GuiLanguage languageGui;
 
-	public GuiSlotLanguage(GuiLanguage var1) {
-		super(var1.mc, var1.width, var1.height, 32, var1.height - 65 + 4, 18);
-		this.languageGui = var1;
-		this.b = Lists.newArrayList();
-		this.h = Maps.newHashMap();
-		Iterator var2 = GuiLanguage.func_135011_a(var1).getLanguages().iterator();
+	public GuiSlotLanguage(GuiLanguage par1GuiLanguage) {
+		super(par1GuiLanguage.mc, par1GuiLanguage.width, par1GuiLanguage.height, 32, par1GuiLanguage.height - 65 + 4, 18);
+		this.languageGui = par1GuiLanguage;
+		this.field_77251_g = Lists.newArrayList();
+		this.field_77253_h = Maps.newHashMap();
+		Iterator var2 = GuiLanguage.func_135011_a(par1GuiLanguage).getLanguages().iterator();
 
-		while(var2.hasNext()) {
+		while (var2.hasNext()) {
 			Language var3 = (Language)var2.next();
-			this.h.put(var3.getLanguageCode(), var3);
-			this.b.add(var3.getLanguageCode());
+			this.field_77253_h.put(var3.getLanguageCode(), var3);
+			this.field_77251_g.add(var3.getLanguageCode());
 		}
-
 	}
 
+	/**
+	 * Gets the size of the current slot list.
+	 */
 	protected int getSize() {
-		return this.b.size();
+		return this.field_77251_g.size();
 	}
 
-	protected void elementClicked(int var1, boolean var2) {
-		Language var3 = (Language)this.h.get(this.b.get(var1));
+	/**
+	 * the element in the slot that was clicked, boolean for wether it was double clicked or not
+	 */
+	protected void elementClicked(int par1, boolean par2) {
+		Language var3 = (Language)this.field_77253_h.get(this.field_77251_g.get(par1));
 		GuiLanguage.func_135011_a(this.languageGui).setCurrentLanguage(var3);
-		GuiLanguage.getGameSettings(this.languageGui).an = var3.getLanguageCode();
+		GuiLanguage.getGameSettings(this.languageGui).language = var3.getLanguageCode();
 		this.languageGui.mc.refreshResources();
 		this.languageGui.fontRenderer.setUnicodeFlag(GuiLanguage.func_135011_a(this.languageGui).isCurrentLocaleUnicode());
 		this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).isCurrentLanguageBidirectional());
-		GuiLanguage.getDoneButton(this.languageGui).f = I18n.getString("gui.done");
+		GuiLanguage.getDoneButton(this.languageGui).displayString = I18n.getString("gui.done");
 		GuiLanguage.getGameSettings(this.languageGui).saveOptions();
 	}
 
-	protected boolean isSelected(int var1) {
-		return ((String)this.b.get(var1)).equals(GuiLanguage.func_135011_a(this.languageGui).getCurrentLanguage().getLanguageCode());
+	/**
+	 * returns true if the element passed in is currently selected
+	 */
+	protected boolean isSelected(int par1) {
+		return ((String)this.field_77251_g.get(par1)).equals(GuiLanguage.func_135011_a(this.languageGui).getCurrentLanguage().getLanguageCode());
 	}
 
+	/**
+	 * return the height of the content being scrolled
+	 */
 	protected int getContentHeight() {
 		return this.getSize() * 18;
 	}
@@ -53,9 +65,9 @@
 		this.languageGui.drawDefaultBackground();
 	}
 
-	protected void drawSlot(int var1, int var2, int var3, int var4, Tessellator var5) {
+	protected void drawSlot(int par1, int par2, int par3, int par4, Tessellator par5Tessellator) {
 		this.languageGui.fontRenderer.setBidiFlag(true);
-		this.languageGui.drawCenteredString(this.languageGui.fontRenderer, ((Language)this.h.get(this.b.get(var1))).toString(), this.languageGui.width / 2, var3 + 1, 16777215);
+		this.languageGui.drawCenteredString(this.languageGui.fontRenderer, ((Language)this.field_77253_h.get(this.field_77251_g.get(par1))).toString(), this.languageGui.width / 2, par3 + 1, 16777215);
 		this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).getCurrentLanguage().isBidirectional());
 	}
 }
