--- net/minecraft/src/IProgressUpdate.java
+++ net/minecraft/src/IProgressUpdate.java
@@ -1,9 +1,19 @@
 package net.minecraft.src;
 
 public interface IProgressUpdate {
-	void displaySavingString(String var1);
-
-	void displayLoadingString(String var1);
-
+
+	/**
+	 * "Saving level", or the loading,or downloading equivelent
+	 */
+	void displayProgressMessage(String var1);
+
+	/**
+	 * This is called with "Working..." by resetProgressAndMessage
+	 */
+	void resetProgresAndWorkingMessage(String var1);
+
+	/**
+	 * Updates the progress bar on the loading screen to the specified amount. Args: loadProgress
+	 */
 	void setLoadingProgress(int var1);
 }
