--- net/minecraft/src/ItemEgg.java
+++ net/minecraft/src/ItemEgg.java
@@ -1,22 +1,26 @@
 package net.minecraft.src;
 
 public class ItemEgg extends Item {
-	public ItemEgg(int var1) {
-		super(var1);
+	public ItemEgg(int par1) {
+		super(par1);
 		this.maxStackSize = 16;
 		this.setCreativeTab(CreativeTabs.tabMaterials);
 	}
 
-	public ItemStack onItemRightClick(ItemStack var1, World var2, EntityPlayer var3) {
-		if(!var3.capabilities.isCreativeMode) {
-			--var1.stackSize;
-		}
-
-		var2.playSoundAtEntity(var3, "random.bow", 0.5F, 0.4F / (f.nextFloat() * 0.4F + 0.8F));
-		if(!var2.isRemote) {
-			var2.spawnEntityInWorld(new EntityEgg(var2, var3));
-		}
-
-		return var1;
+	/**
+	 * Called whenever this item is equipped and the right mouse button is pressed. Args: itemStack, world, entityPlayer
+	 */
+	public ItemStack onItemRightClick(ItemStack par1ItemStack, World par2World, EntityPlayer par3EntityPlayer) {
+		if (!par3EntityPlayer.capabilities.isCreativeMode) {
+			--par1ItemStack.stackSize;
+		}
+
+		par2World.playSoundAtEntity(par3EntityPlayer, "random.bow", 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
+
+		if (!par2World.isRemote) {
+			par2World.spawnEntityInWorld(new EntityEgg(par2World, par3EntityPlayer));
+		}
+
+		return par1ItemStack;
 	}
 }
