--- net/minecraft/src/ItemAppleGold.java
+++ net/minecraft/src/ItemAppleGold.java
@@ -3,38 +3,43 @@
 import java.util.List;
 
 public class ItemAppleGold extends ItemFood {
-	public ItemAppleGold(int var1, int var2, float var3, boolean var4) {
-		super(var1, var2, var3, var4);
+	public ItemAppleGold(int par1, int par2, float par3, boolean par4) {
+		super(par1, par2, par3, par4);
 		this.setHasSubtypes(true);
 	}
 
-	public boolean hasEffect(ItemStack var1) {
-		return var1.getItemDamage() > 0;
-	}
-
-	public EnumRarity getRarity(ItemStack var1) {
-		return var1.getItemDamage() == 0 ? EnumRarity.rare : EnumRarity.epic;
-	}
-
-	protected void onFoodEaten(ItemStack var1, World var2, EntityPlayer var3) {
-		if(!var2.isRemote) {
-			var3.addPotionEffect(new PotionEffect(Potion.field_76444_x.id, 2400, 0));
+	public boolean hasEffect(ItemStack par1ItemStack) {
+		return par1ItemStack.getItemDamage() > 0;
+	}
+
+	/**
+	 * Return an item rarity from EnumRarity
+	 */
+	public EnumRarity getRarity(ItemStack par1ItemStack) {
+		return par1ItemStack.getItemDamage() == 0 ? EnumRarity.rare : EnumRarity.epic;
+	}
+
+	protected void onFoodEaten(ItemStack par1ItemStack, World par2World, EntityPlayer par3EntityPlayer) {
+		if (!par2World.isRemote) {
+			par3EntityPlayer.addPotionEffect(new PotionEffect(Potion.field_76444_x.id, 2400, 0));
 		}
 
-		if(var1.getItemDamage() > 0) {
-			if(!var2.isRemote) {
-				var3.addPotionEffect(new PotionEffect(Potion.regeneration.id, 600, 4));
-				var3.addPotionEffect(new PotionEffect(Potion.resistance.id, 6000, 0));
-				var3.addPotionEffect(new PotionEffect(Potion.fireResistance.id, 6000, 0));
+		if (par1ItemStack.getItemDamage() > 0) {
+			if (!par2World.isRemote) {
+				par3EntityPlayer.addPotionEffect(new PotionEffect(Potion.regeneration.id, 600, 4));
+				par3EntityPlayer.addPotionEffect(new PotionEffect(Potion.resistance.id, 6000, 0));
+				par3EntityPlayer.addPotionEffect(new PotionEffect(Potion.fireResistance.id, 6000, 0));
 			}
 		} else {
-			super.onFoodEaten(var1, var2, var3);
+			super.onFoodEaten(par1ItemStack, par2World, par3EntityPlayer);
 		}
-
 	}
 
-	public void getSubItems(int var1, CreativeTabs var2, List var3) {
-		var3.add(new ItemStack(var1, 1, 0));
-		var3.add(new ItemStack(var1, 1, 1));
+	/**
+	 * returns a list of items with the same ID, but different meta (eg: dye returns 16 items)
+	 */
+	public void getSubItems(int par1, CreativeTabs par2CreativeTabs, List par3List) {
+		par3List.add(new ItemStack(par1, 1, 0));
+		par3List.add(new ItemStack(par1, 1, 1));
 	}
 }
