--- net/minecraft/src/ItemNameTag.java
+++ net/minecraft/src/ItemNameTag.java
@@ -1,22 +1,25 @@
 package net.minecraft.src;
 
 public class ItemNameTag extends Item {
-	public ItemNameTag(int var1) {
-		super(var1);
+	public ItemNameTag(int par1) {
+		super(par1);
 		this.setCreativeTab(CreativeTabs.tabTools);
 	}
 
-	public boolean itemInteractionForEntity(ItemStack var1, EntityPlayer var2, EntityLivingBase var3) {
-		if(!var1.hasDisplayName()) {
+	/**
+	 * Returns true if the item can be used on the given entity, e.g. shears on sheep.
+	 */
+	public boolean itemInteractionForEntity(ItemStack par1ItemStack, EntityPlayer par2EntityPlayer, EntityLivingBase par3EntityLivingBase) {
+		if (!par1ItemStack.hasDisplayName()) {
 			return false;
-		} else if(var3 instanceof EntityLiving) {
-			EntityLiving var4 = (EntityLiving)var3;
-			var4.setCustomNameTag(var1.getDisplayName());
+		} else if (par3EntityLivingBase instanceof EntityLiving) {
+			EntityLiving var4 = (EntityLiving)par3EntityLivingBase;
+			var4.setCustomNameTag(par1ItemStack.getDisplayName());
 			var4.func_110163_bv();
-			--var1.stackSize;
+			--par1ItemStack.stackSize;
 			return true;
 		} else {
-			return super.itemInteractionForEntity(var1, var2, var3);
+			return super.itemInteractionForEntity(par1ItemStack, par2EntityPlayer, par3EntityLivingBase);
 		}
 	}
 }
