--- net/minecraft/src/ItemBucketMilk.java
+++ net/minecraft/src/ItemBucketMilk.java
@@ -1,34 +1,43 @@
 package net.minecraft.src;
 
 public class ItemBucketMilk extends Item {
-	public ItemBucketMilk(int var1) {
-		super(var1);
+	public ItemBucketMilk(int par1) {
+		super(par1);
 		this.setMaxStackSize(1);
 		this.setCreativeTab(CreativeTabs.tabMisc);
 	}
 
-	public ItemStack onEaten(ItemStack var1, World var2, EntityPlayer var3) {
-		if(!var3.capabilities.isCreativeMode) {
-			--var1.stackSize;
-		}
-
-		if(!var2.isRemote) {
-			var3.clearActivePotions();
-		}
-
-		return var1.stackSize <= 0 ? new ItemStack(Item.bucketEmpty) : var1;
+	public ItemStack onEaten(ItemStack par1ItemStack, World par2World, EntityPlayer par3EntityPlayer) {
+		if (!par3EntityPlayer.capabilities.isCreativeMode) {
+			--par1ItemStack.stackSize;
+		}
+
+		if (!par2World.isRemote) {
+			par3EntityPlayer.clearActivePotions();
+		}
+
+		return par1ItemStack.stackSize <= 0 ? new ItemStack(Item.bucketEmpty) : par1ItemStack;
 	}
 
-	public int getMaxItemUseDuration(ItemStack var1) {
+	/**
+	 * How long it takes to use or consume an item
+	 */
+	public int getMaxItemUseDuration(ItemStack par1ItemStack) {
 		return 32;
 	}
 
-	public EnumAction getItemUseAction(ItemStack var1) {
+	/**
+	 * returns the action that specifies what animation to play when the items is being used
+	 */
+	public EnumAction getItemUseAction(ItemStack par1ItemStack) {
 		return EnumAction.drink;
 	}
 
-	public ItemStack onItemRightClick(ItemStack var1, World var2, EntityPlayer var3) {
-		var3.setItemInUse(var1, this.getMaxItemUseDuration(var1));
-		return var1;
+	/**
+	 * Called whenever this item is equipped and the right mouse button is pressed. Args: itemStack, world, entityPlayer
+	 */
+	public ItemStack onItemRightClick(ItemStack par1ItemStack, World par2World, EntityPlayer par3EntityPlayer) {
+		par3EntityPlayer.setItemInUse(par1ItemStack, this.getMaxItemUseDuration(par1ItemStack));
+		return par1ItemStack;
 	}
 }
