--- net/minecraft/src/ItemBlockWithMetadata.java
+++ net/minecraft/src/ItemBlockWithMetadata.java
@@ -3,18 +3,24 @@
 public class ItemBlockWithMetadata extends ItemBlock {
 	private Block theBlock;
 
-	public ItemBlockWithMetadata(int var1, Block var2) {
-		super(var1);
-		this.theBlock = var2;
+	public ItemBlockWithMetadata(int par1, Block par2Block) {
+		super(par1);
+		this.theBlock = par2Block;
 		this.setMaxDamage(0);
 		this.setHasSubtypes(true);
 	}
 
-	public Icon getIconFromDamage(int var1) {
-		return this.theBlock.getIcon(2, var1);
+	/**
+	 * Gets an icon index based on an item's damage value
+	 */
+	public Icon getIconFromDamage(int par1) {
+		return this.theBlock.getIcon(2, par1);
 	}
 
-	public int getMetadata(int var1) {
-		return var1;
+	/**
+	 * Returns the metadata of the block which this Item (ItemBlock) can place
+	 */
+	public int getMetadata(int par1) {
+		return par1;
 	}
 }
