--- net/minecraft/src/ItemCoal.java
+++ net/minecraft/src/ItemCoal.java
@@ -5,28 +5,38 @@
 public class ItemCoal extends Item {
 	private Icon field_111220_a;
 
-	public ItemCoal(int var1) {
-		super(var1);
+	public ItemCoal(int par1) {
+		super(par1);
 		this.setHasSubtypes(true);
 		this.setMaxDamage(0);
 		this.setCreativeTab(CreativeTabs.tabMaterials);
 	}
 
-	public String getUnlocalizedName(ItemStack var1) {
-		return var1.getItemDamage() == 1 ? "item.charcoal" : "item.coal";
-	}
-
-	public void getSubItems(int var1, CreativeTabs var2, List var3) {
-		var3.add(new ItemStack(var1, 1, 0));
-		var3.add(new ItemStack(var1, 1, 1));
-	}
-
-	public Icon getIconFromDamage(int var1) {
-		return var1 == 1 ? this.field_111220_a : super.getIconFromDamage(var1);
-	}
-
-	public void registerIcons(IconRegister var1) {
-		super.registerIcons(var1);
-		this.field_111220_a = var1.registerIcon("charcoal");
+	/**
+	 * Returns the unlocalized name of this item. This version accepts an ItemStack so different stacks can have different
+	 * names based on their damage or NBT.
+	 */
+	public String getUnlocalizedName(ItemStack par1ItemStack) {
+		return par1ItemStack.getItemDamage() == 1 ? "item.charcoal" : "item.coal";
+	}
+
+	/**
+	 * returns a list of items with the same ID, but different meta (eg: dye returns 16 items)
+	 */
+	public void getSubItems(int par1, CreativeTabs par2CreativeTabs, List par3List) {
+		par3List.add(new ItemStack(par1, 1, 0));
+		par3List.add(new ItemStack(par1, 1, 1));
+	}
+
+	/**
+	 * Gets an icon index based on an item's damage value
+	 */
+	public Icon getIconFromDamage(int par1) {
+		return par1 == 1 ? this.field_111220_a : super.getIconFromDamage(par1);
+	}
+
+	public void registerIcons(IconRegister par1IconRegister) {
+		super.registerIcons(par1IconRegister);
+		this.field_111220_a = par1IconRegister.registerIcon("charcoal");
 	}
 }
