--- net/minecraft/src/MainProxyAuthenticator.java
+++ net/minecraft/src/MainProxyAuthenticator.java
@@ -4,15 +4,16 @@
 import java.net.PasswordAuthentication;
 
 public final class MainProxyAuthenticator extends Authenticator {
-	final String a;
-	final String b;
-
-	public MainProxyAuthenticator(String var1, String var2) {
-		this.a = var1;
-		this.b = var2;
+	final String field_111237_a;
+
+	final String field_111236_b;
+
+	public MainProxyAuthenticator(String par1Str, String par2Str) {
+		this.field_111237_a = par1Str;
+		this.field_111236_b = par2Str;
 	}
 
 	protected PasswordAuthentication getPasswordAuthentication() {
-		return new PasswordAuthentication(this.a, this.b.toCharArray());
+		return new PasswordAuthentication(this.field_111237_a, this.field_111236_b.toCharArray());
 	}
 }
