--- net/minecraft/src/MapGenStructureData.java
+++ net/minecraft/src/MapGenStructureData.java
@@ -3,26 +3,32 @@
 public class MapGenStructureData extends WorldSavedData {
 	private NBTTagCompound field_143044_a = new NBTTagCompound("Features");
 
-	public MapGenStructureData(String var1) {
-		super(var1);
-	}
-
-	public void readFromNBT(NBTTagCompound var1) {
-		this.field_143044_a = var1.getCompoundTag("Features");
-	}
-
-	public void writeToNBT(NBTTagCompound var1) {
-		var1.setTag("Features", this.field_143044_a);
-	}
-
-	public void func_143043_a(NBTTagCompound var1, int var2, int var3) {
-		String var4 = this.func_143042_b(var2, var3);
-		var1.setName(var4);
-		this.field_143044_a.setTag(var4, var1);
-	}
-
-	public String func_143042_b(int var1, int var2) {
-		return "[" + var1 + "," + var2 + "]";
+	public MapGenStructureData(String par1Str) {
+		super(par1Str);
+	}
+
+	/**
+	 * reads in data from the NBTTagCompound into this MapDataBase
+	 */
+	public void readFromNBT(NBTTagCompound par1NBTTagCompound) {
+		this.field_143044_a = par1NBTTagCompound.getCompoundTag("Features");
+	}
+
+	/**
+	 * write data to NBTTagCompound from this MapDataBase, similar to Entities and TileEntities
+	 */
+	public void writeToNBT(NBTTagCompound par1NBTTagCompound) {
+		par1NBTTagCompound.setTag("Features", this.field_143044_a);
+	}
+
+	public void func_143043_a(NBTTagCompound par1NBTTagCompound, int par2, int par3) {
+		String var4 = this.func_143042_b(par2, par3);
+		par1NBTTagCompound.setName(var4);
+		this.field_143044_a.setTag(var4, par1NBTTagCompound);
+	}
+
+	public String func_143042_b(int par1, int par2) {
+		return "[" + par1 + "," + par2 + "]";
 	}
 
 	public NBTTagCompound func_143041_a() {
