--- net/minecraft/src/McoServerListUpdateTaskComparator.java
+++ net/minecraft/src/McoServerListUpdateTaskComparator.java
@@ -3,41 +3,42 @@
 import java.util.Comparator;
 
 class McoServerListUpdateTaskComparator implements Comparator {
-	private final String b;
+	private final String field_140069_b;
+
 	final McoServerListUpdateTask field_140070_a;
 
-	private McoServerListUpdateTaskComparator(McoServerListUpdateTask var1, String var2) {
-		this.field_140070_a = var1;
-		this.b = var2;
+	private McoServerListUpdateTaskComparator(McoServerListUpdateTask par1McoServerListUpdateTask, String par2Str) {
+		this.field_140070_a = par1McoServerListUpdateTask;
+		this.field_140069_b = par2Str;
 	}
 
-	public int func_140068_a(McoServer var1, McoServer var2) {
-		if(var1.e.equals(var2.e)) {
-			return var1.field_96408_a < var2.field_96408_a ? 1 : (var1.field_96408_a > var2.field_96408_a ? -1 : 0);
-		} else if(var1.e.equals(this.b)) {
+	public int func_140068_a(McoServer par1McoServer, McoServer par2McoServer) {
+		if (par1McoServer.field_96405_e.equals(par2McoServer.field_96405_e)) {
+			return par1McoServer.field_96408_a < par2McoServer.field_96408_a ? 1 : (par1McoServer.field_96408_a > par2McoServer.field_96408_a ? -1 : 0);
+		} else if (par1McoServer.field_96405_e.equals(this.field_140069_b)) {
 			return -1;
-		} else if(var2.e.equals(this.b)) {
+		} else if (par2McoServer.field_96405_e.equals(this.field_140069_b)) {
 			return 1;
 		} else {
-			if(var1.d.equals("CLOSED") || var2.d.equals("CLOSED")) {
-				if(var1.d.equals("CLOSED")) {
+			if (par1McoServer.field_96404_d.equals("CLOSED") || par2McoServer.field_96404_d.equals("CLOSED")) {
+				if (par1McoServer.field_96404_d.equals("CLOSED")) {
 					return 1;
 				}
 
-				if(var2.d.equals("CLOSED")) {
+				if (par2McoServer.field_96404_d.equals("CLOSED")) {
 					return 0;
 				}
 			}
 
-			return var1.field_96408_a < var2.field_96408_a ? 1 : (var1.field_96408_a > var2.field_96408_a ? -1 : 0);
+			return par1McoServer.field_96408_a < par2McoServer.field_96408_a ? 1 : (par1McoServer.field_96408_a > par2McoServer.field_96408_a ? -1 : 0);
 		}
 	}
 
-	public int compare(Object var1, Object var2) {
-		return this.func_140068_a((McoServer)var1, (McoServer)var2);
+	public int compare(Object par1Obj, Object par2Obj) {
+		return this.func_140068_a((McoServer)par1Obj, (McoServer)par2Obj);
 	}
 
-	McoServerListUpdateTaskComparator(McoServerListUpdateTask var1, String var2, McoServerListEmptyAnon var3) {
-		this(var1, var2);
+	McoServerListUpdateTaskComparator(McoServerListUpdateTask par1McoServerListUpdateTask, String par2Str, McoServerListEmptyAnon par3McoServerListEmptyAnon) {
+		this(par1McoServerListUpdateTask, par2Str);
 	}
 }
