--- net/minecraft/src/ModelChest.java
+++ net/minecraft/src/ModelChest.java
@@ -1,8 +1,14 @@
 package net.minecraft.src;
 
 public class ModelChest extends ModelBase {
+
+	/** The chest lid in the chest's model. */
 	public ModelRenderer chestLid = (new ModelRenderer(this, 0, 0)).setTextureSize(64, 64);
+
+	/** The model of the bottom of the chest. */
 	public ModelRenderer chestBelow;
+
+	/** The chest's knob in the chest model. */
 	public ModelRenderer chestKnob;
 
 	public ModelChest() {
@@ -22,10 +28,13 @@
 		this.chestBelow.rotationPointZ = 1.0F;
 	}
 
+	/**
+	 * This method renders out all parts of the chest model.
+	 */
 	public void renderAll() {
 		this.chestKnob.rotateAngleX = this.chestLid.rotateAngleX;
-		this.chestLid.render(1.0F / 16.0F);
-		this.chestKnob.render(1.0F / 16.0F);
-		this.chestBelow.render(1.0F / 16.0F);
+		this.chestLid.render(0.0625F);
+		this.chestKnob.render(0.0625F);
+		this.chestBelow.render(0.0625F);
 	}
 }
