--- net/minecraft/src/ModelWitch.java
+++ net/minecraft/src/ModelWitch.java
@@ -5,8 +5,8 @@
 	private ModelRenderer field_82901_h = (new ModelRenderer(this)).setTextureSize(64, 128);
 	private ModelRenderer witchHat;
 
-	public ModelWitch(float var1) {
-		super(var1, 0.0F, 64, 128);
+	public ModelWitch(float par1) {
+		super(par1, 0.0F, 64, 128);
 		this.field_82901_h.setRotationPoint(0.0F, -2.0F, 0.0F);
 		this.field_82901_h.setTextureOffset(0, 0).addBox(0.0F, 3.0F, -6.75F, 1, 1, 1, -0.25F);
 		this.villagerNose.addChild(this.field_82901_h);
@@ -34,18 +34,23 @@
 		var3.addChild(var4);
 	}
 
-	public void setRotationAngles(float var1, float var2, float var3, float var4, float var5, float var6, Entity var7) {
-		super.setRotationAngles(var1, var2, var3, var4, var5, var6, var7);
+	/**
+	 * Sets the model's various rotation angles. For bipeds, par1 and par2 are used for animating the movement of arms and
+	 * legs, where par1 represents the time(so that arms and legs swing back and forth) and par2 represents how "far" arms
+	 * and legs can swing at most.
+	 */
+	public void setRotationAngles(float par1, float par2, float par3, float par4, float par5, float par6, Entity par7Entity) {
+		super.setRotationAngles(par1, par2, par3, par4, par5, par6, par7Entity);
 		this.villagerNose.offsetX = this.villagerNose.offsetY = this.villagerNose.offsetZ = 0.0F;
-		float var8 = 0.01F * (float)(var7.entityId % 10);
-		this.villagerNose.rotateAngleX = MathHelper.sin((float)var7.ticksExisted * var8) * 4.5F * (float)Math.PI / 180.0F;
+		float var8 = 0.01F * (float)(par7Entity.entityId % 10);
+		this.villagerNose.rotateAngleX = MathHelper.sin((float)par7Entity.ticksExisted * var8) * 4.5F * (float)Math.PI / 180.0F;
 		this.villagerNose.rotateAngleY = 0.0F;
-		this.villagerNose.rotateAngleZ = MathHelper.cos((float)var7.ticksExisted * var8) * 2.5F * (float)Math.PI / 180.0F;
-		if(this.field_82900_g) {
+		this.villagerNose.rotateAngleZ = MathHelper.cos((float)par7Entity.ticksExisted * var8) * 2.5F * (float)Math.PI / 180.0F;
+
+		if (this.field_82900_g) {
 			this.villagerNose.rotateAngleX = -0.9F;
 			this.villagerNose.offsetZ = -0.09375F;
-			this.villagerNose.offsetY = 3.0F / 16.0F;
+			this.villagerNose.offsetY = 0.1875F;
 		}
-
 	}
 }
