--- net/minecraft/src/ModelSign.java
+++ net/minecraft/src/ModelSign.java
@@ -1,7 +1,11 @@
 package net.minecraft.src;
 
 public class ModelSign extends ModelBase {
+
+	/** The board on a sign that has the writing on it. */
 	public ModelRenderer signBoard = new ModelRenderer(this, 0, 0);
+
+	/** The stick a sign stands on. */
 	public ModelRenderer signStick;
 
 	public ModelSign() {
@@ -10,8 +14,11 @@
 		this.signStick.addBox(-1.0F, -2.0F, -1.0F, 2, 14, 2, 0.0F);
 	}
 
+	/**
+	 * Renders the sign model through TileEntitySignRenderer
+	 */
 	public void renderSign() {
-		this.signBoard.render(1.0F / 16.0F);
-		this.signStick.render(1.0F / 16.0F);
+		this.signBoard.render(0.0625F);
+		this.signStick.render(0.0625F);
 	}
 }
