--- net/minecraft/src/PendingInvite.java
+++ net/minecraft/src/PendingInvite.java
@@ -3,18 +3,19 @@
 import argo.jdom.JsonNode;
 
 public class PendingInvite extends ValueObject {
-	public String a;
-	public String b;
-	public String c;
+	public String field_130094_a;
+	public String field_130092_b;
+	public String field_130093_c;
 
-	public static PendingInvite func_130091_a(JsonNode var0) {
+	public static PendingInvite func_130091_a(JsonNode par0JsonNode) {
 		PendingInvite var1 = new PendingInvite();
 
 		try {
-			var1.a = var0.getStringValue(new Object[]{"invitationId"});
-			var1.b = var0.getStringValue(new Object[]{"worldName"});
-			var1.c = var0.getStringValue(new Object[]{"worldOwnerName"});
+			var1.field_130094_a = par0JsonNode.getStringValue(new Object[] {"invitationId"});
+			var1.field_130092_b = par0JsonNode.getStringValue(new Object[] {"worldName"});
+			var1.field_130093_c = par0JsonNode.getStringValue(new Object[] {"worldOwnerName"});
 		} catch (Exception var3) {
+			;
 		}
 
 		return var1;
