--- net/minecraft/src/PlayerUsageSnooperThread.java
+++ net/minecraft/src/PlayerUsageSnooperThread.java
@@ -1,26 +1,27 @@
 package net.minecraft.src;
 
 import java.util.HashMap;
-import java.util.Map;
 import java.util.TimerTask;
 
 class PlayerUsageSnooperThread extends TimerTask {
+
+	/** The PlayerUsageSnooper object. */
 	final PlayerUsageSnooper snooper;
 
-	PlayerUsageSnooperThread(PlayerUsageSnooper var1) {
-		this.snooper = var1;
+	PlayerUsageSnooperThread(PlayerUsageSnooper par1PlayerUsageSnooper) {
+		this.snooper = par1PlayerUsageSnooper;
 	}
 
 	public void run() {
-		if(PlayerUsageSnooper.getStatsCollectorFor(this.snooper).isSnooperEnabled()) {
-			Object var2 = PlayerUsageSnooper.getSyncLockFor(this.snooper);
+		if (PlayerUsageSnooper.getStatsCollectorFor(this.snooper).isSnooperEnabled()) {
 			HashMap var1;
-			synchronized(var2) {
+
+			synchronized (PlayerUsageSnooper.getSyncLockFor(this.snooper)) {
 				var1 = new HashMap(PlayerUsageSnooper.getDataMapFor(this.snooper));
 				var1.put("snooper_count", Integer.valueOf(PlayerUsageSnooper.getSelfCounterFor(this.snooper)));
 			}
 
-			HttpUtil.sendPost(PlayerUsageSnooper.getStatsCollectorFor(this.snooper).getLogAgent(), PlayerUsageSnooper.getServerUrlFor(this.snooper), (Map)var1, true);
+			HttpUtil.sendPost(PlayerUsageSnooper.getStatsCollectorFor(this.snooper).getLogAgent(), PlayerUsageSnooper.getServerUrlFor(this.snooper), var1, true);
 		}
 	}
 }
