--- net/minecraft/src/PotionHealthBoost.java
+++ net/minecraft/src/PotionHealthBoost.java
@@ -1,15 +1,15 @@
 package net.minecraft.src;
 
 public class PotionHealthBoost extends Potion {
-	public PotionHealthBoost(int var1, boolean var2, int var3) {
-		super(var1, var2, var3);
+	public PotionHealthBoost(int par1, boolean par2, int par3) {
+		super(par1, par2, par3);
 	}
 
-	public void removeAttributesModifiersFromEntity(EntityLivingBase var1, BaseAttributeMap var2, int var3) {
-		super.removeAttributesModifiersFromEntity(var1, var2, var3);
-		if(var1.getHealth() > var1.getMaxHealth()) {
-			var1.setHealth(var1.getMaxHealth());
+	public void removeAttributesModifiersFromEntity(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3) {
+		super.removeAttributesModifiersFromEntity(par1EntityLivingBase, par2BaseAttributeMap, par3);
+
+		if (par1EntityLivingBase.getHealth() > par1EntityLivingBase.getMaxHealth()) {
+			par1EntityLivingBase.setHealth(par1EntityLivingBase.getMaxHealth());
 		}
-
 	}
 }
