--- net/minecraft/src/RecipesIngots.java
+++ net/minecraft/src/RecipesIngots.java
@@ -1,17 +1,24 @@
 package net.minecraft.src;
 
 public class RecipesIngots {
-	private Object[][] a = new Object[][]{{Block.blockGold, new ItemStack(Item.ingotGold, 9)}, {Block.blockIron, new ItemStack(Item.ingotIron, 9)}, {Block.blockDiamond, new ItemStack(Item.diamond, 9)}, {Block.blockEmerald, new ItemStack(Item.emerald, 9)}, {Block.blockLapis, new ItemStack(Item.dyePowder, 9, 4)}, {Block.blockRedstone, new ItemStack(Item.redstone, 9)}, {Block.coalBlock, new ItemStack(Item.coal, 9, 0)}, {Block.hay, new ItemStack(Item.wheat, 9)}};
-
-	public void addRecipes(CraftingManager var1) {
-		for(int var2 = 0; var2 < this.a.length; ++var2) {
-			Block var3 = (Block)this.a[var2][0];
-			ItemStack var4 = (ItemStack)this.a[var2][1];
-			var1.addRecipe(new ItemStack(var3), new Object[]{"###", "###", "###", Character.valueOf('#'), var4});
-			var1.addRecipe(var4, new Object[]{"#", Character.valueOf('#'), var3});
+	private Object[][] recipeItems;
+
+	public RecipesIngots() {
+		this.recipeItems = new Object[][] {{Block.blockGold, new ItemStack(Item.ingotGold, 9)}, {Block.blockIron, new ItemStack(Item.ingotIron, 9)}, {Block.blockDiamond, new ItemStack(Item.diamond, 9)}, {Block.blockEmerald, new ItemStack(Item.emerald, 9)}, {Block.blockLapis, new ItemStack(Item.dyePowder, 9, 4)}, {Block.blockRedstone, new ItemStack(Item.redstone, 9)}, {Block.coalBlock, new ItemStack(Item.coal, 9, 0)}, {Block.hay, new ItemStack(Item.wheat, 9)}};
+	}
+
+	/**
+	 * Adds the ingot recipes to the CraftingManager.
+	 */
+	public void addRecipes(CraftingManager par1CraftingManager) {
+		for (int var2 = 0; var2 < this.recipeItems.length; ++var2) {
+			Block var3 = (Block)this.recipeItems[var2][0];
+			ItemStack var4 = (ItemStack)this.recipeItems[var2][1];
+			par1CraftingManager.addRecipe(new ItemStack(var3), new Object[] {"###", "###", "###", '#', var4});
+			par1CraftingManager.addRecipe(var4, new Object[] {"#", '#', var3});
 		}
 
-		var1.addRecipe(new ItemStack(Item.ingotGold), new Object[]{"###", "###", "###", Character.valueOf('#'), Item.goldNugget});
-		var1.addRecipe(new ItemStack(Item.goldNugget, 9), new Object[]{"#", Character.valueOf('#'), Item.ingotGold});
+		par1CraftingManager.addRecipe(new ItemStack(Item.ingotGold), new Object[] {"###", "###", "###", '#', Item.goldNugget});
+		par1CraftingManager.addRecipe(new ItemStack(Item.goldNugget, 9), new Object[] {"#", '#', Item.ingotGold});
 	}
 }
