--- net/minecraft/src/RenderFireball.java
+++ net/minecraft/src/RenderFireball.java
@@ -6,14 +6,14 @@
 public class RenderFireball extends Render {
 	private float field_77002_a;
 
-	public RenderFireball(float var1) {
-		this.field_77002_a = var1;
+	public RenderFireball(float par1) {
+		this.field_77002_a = par1;
 	}
 
-	public void doRenderFireball(EntityFireball var1, double var2, double var4, double var6, float var8, float var9) {
+	public void doRenderFireball(EntityFireball par1EntityFireball, double par2, double par4, double par6, float par8, float par9) {
 		GL11.glPushMatrix();
-		this.bindEntityTexture(var1);
-		GL11.glTranslatef((float)var2, (float)var4, (float)var6);
+		this.bindEntityTexture(par1EntityFireball);
+		GL11.glTranslatef((float)par2, (float)par4, (float)par6);
 		GL11.glEnable(GL12.GL_RESCALE_NORMAL);
 		float var10 = this.field_77002_a;
 		GL11.glScalef(var10 / 1.0F, var10 / 1.0F, var10 / 1.0F);
@@ -39,15 +39,24 @@
 		GL11.glPopMatrix();
 	}
 
-	protected ResourceLocation getFireballTextures(EntityFireball var1) {
+	protected ResourceLocation getFireballTextures(EntityFireball par1EntityFireball) {
 		return TextureMap.locationItemsTexture;
 	}
 
-	protected ResourceLocation getEntityTexture(Entity var1) {
-		return this.getFireballTextures((EntityFireball)var1);
+	/**
+	 * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
+	 */
+	protected ResourceLocation getEntityTexture(Entity par1Entity) {
+		return this.getFireballTextures((EntityFireball)par1Entity);
 	}
 
-	public void doRender(Entity var1, double var2, double var4, double var6, float var8, float var9) {
-		this.doRenderFireball((EntityFireball)var1, var2, var4, var6, var8, var9);
+	/**
+	 * Actually renders the given argument. This is a synthetic bridge method, always casting down its argument and then
+	 * handing it off to a worker function which does the actual work. In all probabilty, the class Render is generic
+	 * (Render<T extends Entity) and this method has signature public void doRender(T entity, double d, double d1,
+	 * double d2, float f, float f1). But JAD is pre 1.5 so doesn't do that.
+	 */
+	public void doRender(Entity par1Entity, double par2, double par4, double par6, float par8, float par9) {
+		this.doRenderFireball((EntityFireball)par1Entity, par2, par4, par6, par8, par9);
 	}
 }
