--- net/minecraft/src/ResourcePackFileNotFoundException.java
+++ net/minecraft/src/ResourcePackFileNotFoundException.java
@@ -4,7 +4,7 @@
 import java.io.FileNotFoundException;
 
 public class ResourcePackFileNotFoundException extends FileNotFoundException {
-	public ResourcePackFileNotFoundException(File var1, String var2) {
-		super(String.format("\'%s\' in ResourcePack \'%s\'", new Object[]{var2, var1}));
+	public ResourcePackFileNotFoundException(File par1File, String par2Str) {
+		super(String.format("\'%s\' in ResourcePack \'%s\'", new Object[] {par2Str, par1File}));
 	}
 }
