--- net/minecraft/src/ResourcePackRepositoryFilter.java
+++ net/minecraft/src/ResourcePackRepositoryFilter.java
@@ -4,9 +4,9 @@
 import java.io.FileFilter;
 
 final class ResourcePackRepositoryFilter implements FileFilter {
-	public boolean accept(File var1) {
-		boolean var2 = var1.isFile() && var1.getName().endsWith(".zip");
-		boolean var3 = var1.isDirectory() && (new File(var1, "pack.mcmeta")).isFile();
+	public boolean accept(File par1File) {
+		boolean var2 = par1File.isFile() && par1File.getName().endsWith(".zip");
+		boolean var3 = par1File.isDirectory() && (new File(par1File, "pack.mcmeta")).isFile();
 		return var2 || var3;
 	}
 }
