--- net/minecraft/src/RunnableTitleScreen.java
+++ net/minecraft/src/RunnableTitleScreen.java
@@ -5,18 +5,19 @@
 class RunnableTitleScreen extends Thread {
 	final GuiMainMenu theMainMenu;
 
-	RunnableTitleScreen(GuiMainMenu var1) {
-		this.theMainMenu = var1;
+	RunnableTitleScreen(GuiMainMenu par1GuiMainMenu) {
+		this.theMainMenu = par1GuiMainMenu;
 	}
 
 	public void run() {
 		McoClient var1 = new McoClient(GuiMainMenu.func_110348_a(this.theMainMenu).getSession());
 		boolean var2 = false;
 
-		for(int var3 = 0; var3 < 3; ++var3) {
+		for (int var3 = 0; var3 < 3; ++var3) {
 			try {
 				Boolean var4 = var1.func_96375_b();
-				if(var4.booleanValue()) {
+
+				if (var4.booleanValue()) {
 					GuiMainMenu.func_130021_b(this.theMainMenu);
 				}
 
@@ -29,7 +30,7 @@
 				GuiMainMenu.func_130019_d(this.theMainMenu).getLogAgent().logWarning("Realms: could not parse response");
 			}
 
-			if(!var2) {
+			if (!var2) {
 				break;
 			}
 
@@ -39,6 +40,5 @@
 				Thread.currentThread().interrupt();
 			}
 		}
-
 	}
 }
