--- net/minecraft/src/ScheduledSound.java
+++ net/minecraft/src/ScheduledSound.java
@@ -1,7 +1,7 @@
 package net.minecraft.src;
 
 class ScheduledSound {
-	String a;
+	String field_92069_a;
 	float field_92067_b;
 	float field_92068_c;
 	float field_92065_d;
@@ -9,13 +9,13 @@
 	float field_92063_f;
 	int field_92064_g;
 
-	public ScheduledSound(String var1, float var2, float var3, float var4, float var5, float var6, int var7) {
-		this.a = var1;
-		this.field_92067_b = var2;
-		this.field_92068_c = var3;
-		this.field_92065_d = var4;
-		this.field_92066_e = var5;
-		this.field_92063_f = var6;
-		this.field_92064_g = var7;
+	public ScheduledSound(String par1Str, float par2, float par3, float par4, float par5, float par6, int par7) {
+		this.field_92069_a = par1Str;
+		this.field_92067_b = par2;
+		this.field_92068_c = par3;
+		this.field_92065_d = par4;
+		this.field_92066_e = par5;
+		this.field_92063_f = par6;
+		this.field_92064_g = par7;
 	}
 }
