--- net/minecraft/src/SimpleReloadableResourceManagerINNER1.java
+++ net/minecraft/src/SimpleReloadableResourceManagerINNER1.java
@@ -5,15 +5,15 @@
 class SimpleReloadableResourceManagerINNER1 implements Function {
 	final SimpleReloadableResourceManager theSimpleReloadableResourceManager;
 
-	SimpleReloadableResourceManagerINNER1(SimpleReloadableResourceManager var1) {
-		this.theSimpleReloadableResourceManager = var1;
-	}
-
-	public String apply(ResourcePack var1) {
-		return var1.getPackName();
-	}
-
-	public Object apply(Object var1) {
-		return this.apply((ResourcePack)var1);
+	SimpleReloadableResourceManagerINNER1(SimpleReloadableResourceManager par1SimpleReloadableResourceManager) {
+		this.theSimpleReloadableResourceManager = par1SimpleReloadableResourceManager;
+	}
+
+	public String apply(ResourcePack par1ResourcePack) {
+		return par1ResourcePack.getPackName();
+	}
+
+	public Object apply(Object par1Obj) {
+		return this.apply((ResourcePack)par1Obj);
 	}
 }
