--- net/minecraft/src/SimpleTexture.java
+++ net/minecraft/src/SimpleTexture.java
@@ -1,5 +1,6 @@
 package net.minecraft.src;
 
+import com.prupe.mcpatcher.hd.MipmapHelper;
 import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.io.InputStream;
@@ -8,23 +9,25 @@
 public class SimpleTexture extends AbstractTexture {
 	private final ResourceLocation textureLocation;
 
-	public SimpleTexture(ResourceLocation var1) {
-		this.textureLocation = var1;
+	public SimpleTexture(ResourceLocation par1ResourceLocation) {
+		this.textureLocation = par1ResourceLocation;
 	}
 
-	public void loadTexture(ResourceManager var1) throws IOException {
+	public void loadTexture(ResourceManager par1ResourceManager) throws IOException {
 		InputStream var2 = null;
 
 		try {
-			Resource var3 = var1.getResource(this.textureLocation);
+			Resource var3 = par1ResourceManager.getResource(this.textureLocation);
 			var2 = var3.getInputStream();
 			BufferedImage var4 = ImageIO.read(var2);
 			boolean var5 = false;
 			boolean var6 = false;
-			if(var3.hasMetadata()) {
+
+			if (var3.hasMetadata()) {
 				try {
 					TextureMetadataSection var7 = (TextureMetadataSection)var3.getMetadata("texture");
-					if(var7 != null) {
+
+					if (var7 != null) {
 						var5 = var7.getTextureBlur();
 						var6 = var7.getTextureClamp();
 					}
@@ -33,13 +36,11 @@
 				}
 			}
 
-			TextureUtil.uploadTextureImageAllocate(this.getGlTextureId(), var4, var5, var6);
+			MipmapHelper.setupTexture(this.getGlTextureId(), var4, var5, var6, this.textureLocation);
 		} finally {
-			if(var2 != null) {
+			if (var2 != null) {
 				var2.close();
 			}
-
 		}
-
 	}
 }
