--- net/minecraft/src/SlotEnchantment.java
+++ net/minecraft/src/SlotEnchantment.java
@@ -1,14 +1,19 @@
 package net.minecraft.src;
 
 class SlotEnchantment extends Slot {
+
+	/** The brewing stand this slot belongs to. */
 	final ContainerEnchantment container;
 
-	SlotEnchantment(ContainerEnchantment var1, IInventory var2, int var3, int var4, int var5) {
-		super(var2, var3, var4, var5);
-		this.container = var1;
+	SlotEnchantment(ContainerEnchantment par1ContainerEnchantment, IInventory par2IInventory, int par3, int par4, int par5) {
+		super(par2IInventory, par3, par4, par5);
+		this.container = par1ContainerEnchantment;
 	}
 
-	public boolean isItemValid(ItemStack var1) {
+	/**
+	 * Check if the stack is a valid item for this slot. Always true beside for the armor slots.
+	 */
+	public boolean isItemValid(ItemStack par1ItemStack) {
 		return true;
 	}
 }
