--- net/minecraft/src/SlotEnchantmentTable.java
+++ net/minecraft/src/SlotEnchantmentTable.java
@@ -1,23 +1,35 @@
 package net.minecraft.src;
 
 class SlotEnchantmentTable extends InventoryBasic {
+
+	/** The brewing stand this slot belongs to. */
 	final ContainerEnchantment container;
 
-	SlotEnchantmentTable(ContainerEnchantment var1, String var2, boolean var3, int var4) {
-		super(var2, var3, var4);
-		this.container = var1;
+	SlotEnchantmentTable(ContainerEnchantment par1ContainerEnchantment, String par2Str, boolean par3, int par4) {
+		super(par2Str, par3, par4);
+		this.container = par1ContainerEnchantment;
 	}
 
+	/**
+	 * Returns the maximum stack size for a inventory slot. Seems to always be 64, possibly will be extended. *Isn't this
+	 * more of a set than a get?*
+	 */
 	public int getInventoryStackLimit() {
 		return 1;
 	}
 
+	/**
+	 * Called when an the contents of an Inventory change, usually
+	 */
 	public void onInventoryChanged() {
 		super.onInventoryChanged();
 		this.container.onCraftMatrixChanged(this);
 	}
 
-	public boolean isItemValidForSlot(int var1, ItemStack var2) {
+	/**
+	 * Returns true if automation is allowed to insert the given stack (ignoring stack size) into the given slot.
+	 */
+	public boolean isItemValidForSlot(int par1, ItemStack par2ItemStack) {
 		return true;
 	}
 }
