--- net/minecraft/src/SpawnListEntry.java
+++ net/minecraft/src/SpawnListEntry.java
@@ -1,18 +1,20 @@
 package net.minecraft.src;
 
 public class SpawnListEntry extends WeightedRandomItem {
-	public Class b;
+
+	/** Holds the class of the entity to be spawned. */
+	public Class entityClass;
 	public int minGroupCount;
 	public int maxGroupCount;
 
-	public SpawnListEntry(Class var1, int var2, int var3, int var4) {
-		super(var2);
-		this.b = var1;
-		this.minGroupCount = var3;
-		this.maxGroupCount = var4;
+	public SpawnListEntry(Class par1Class, int par2, int par3, int par4) {
+		super(par2);
+		this.entityClass = par1Class;
+		this.minGroupCount = par3;
+		this.maxGroupCount = par4;
 	}
 
 	public String toString() {
-		return this.b.getSimpleName() + "*(" + this.minGroupCount + "-" + this.maxGroupCount + "):" + this.itemWeight;
+		return this.entityClass.getSimpleName() + "*(" + this.minGroupCount + "-" + this.maxGroupCount + "):" + this.itemWeight;
 	}
 }
