--- net/minecraft/src/StatTypeTime.java
+++ net/minecraft/src/StatTypeTime.java
@@ -1,8 +1,12 @@
 package net.minecraft.src;
 
 final class StatTypeTime implements IStatType {
-	public String format(int var1) {
-		double var2 = (double)var1 / 20.0D;
+
+	/**
+	 * Formats a given stat for human consumption.
+	 */
+	public String format(int par1) {
+		double var2 = (double)par1 / 20.0D;
 		double var4 = var2 / 60.0D;
 		double var6 = var4 / 60.0D;
 		double var8 = var6 / 24.0D;
