--- net/minecraft/src/StepSoundSand.java
+++ net/minecraft/src/StepSoundSand.java
@@ -1,10 +1,13 @@
 package net.minecraft.src;
 
 final class StepSoundSand extends StepSound {
-	StepSoundSand(String var1, float var2, float var3) {
-		super(var1, var2, var3);
+	StepSoundSand(String par1Str, float par2, float par3) {
+		super(par1Str, par2, par3);
 	}
 
+	/**
+	 * Used when a block breaks, EXA: Player break, Shep eating grass, etc..
+	 */
 	public String getBreakSound() {
 		return "dig.wood";
 	}
