--- net/minecraft/src/StepSound.java
+++ net/minecraft/src/StepSound.java
@@ -1,14 +1,14 @@
 package net.minecraft.src;
 
 public class StepSound {
-	public final String a;
+	public final String stepSoundName;
 	public final float stepSoundVolume;
 	public final float stepSoundPitch;
 
-	public StepSound(String var1, float var2, float var3) {
-		this.a = var1;
-		this.stepSoundVolume = var2;
-		this.stepSoundPitch = var3;
+	public StepSound(String par1Str, float par2, float par3) {
+		this.stepSoundName = par1Str;
+		this.stepSoundVolume = par2;
+		this.stepSoundPitch = par3;
 	}
 
 	public float getVolume() {
@@ -19,14 +19,23 @@
 		return this.stepSoundPitch;
 	}
 
+	/**
+	 * Used when a block breaks, EXA: Player break, Shep eating grass, etc..
+	 */
 	public String getBreakSound() {
-		return "dig." + this.a;
+		return "dig." + this.stepSoundName;
 	}
 
+	/**
+	 * Used when a entity walks over, or otherwise interacts with the block.
+	 */
 	public String getStepSound() {
-		return "step." + this.a;
+		return "step." + this.stepSoundName;
 	}
 
+	/**
+	 * Used when a player places a block.
+	 */
 	public String getPlaceSound() {
 		return this.getBreakSound();
 	}
