--- net/minecraft/src/StepSoundStone.java
+++ net/minecraft/src/StepSoundStone.java
@@ -1,14 +1,20 @@
 package net.minecraft.src;
 
 final class StepSoundStone extends StepSound {
-	StepSoundStone(String var1, float var2, float var3) {
-		super(var1, var2, var3);
+	StepSoundStone(String par1Str, float par2, float par3) {
+		super(par1Str, par2, par3);
 	}
 
+	/**
+	 * Used when a block breaks, EXA: Player break, Shep eating grass, etc..
+	 */
 	public String getBreakSound() {
 		return "random.glass";
 	}
 
+	/**
+	 * Used when a player places a block.
+	 */
 	public String getPlaceSound() {
 		return "step.stone";
 	}
