--- net/minecraft/src/StructureStrongholdPieceWeight2.java
+++ net/minecraft/src/StructureStrongholdPieceWeight2.java
@@ -1,11 +1,11 @@
 package net.minecraft.src;
 
 final class StructureStrongholdPieceWeight2 extends StructureStrongholdPieceWeight {
-	StructureStrongholdPieceWeight2(Class var1, int var2, int var3) {
-		super(var1, var2, var3);
+	StructureStrongholdPieceWeight2(Class par1Class, int par2, int par3) {
+		super(par1Class, par2, par3);
 	}
 
-	public boolean canSpawnMoreStructuresOfType(int var1) {
-		return super.canSpawnMoreStructuresOfType(var1) && var1 > 4;
+	public boolean canSpawnMoreStructuresOfType(int par1) {
+		return super.canSpawnMoreStructuresOfType(par1) && par1 > 4;
 	}
 }
