--- net/minecraft/src/TileEntitySpecialRenderer.java
+++ net/minecraft/src/TileEntitySpecialRenderer.java
@@ -1,24 +1,34 @@
 package net.minecraft.src;
 
 public abstract class TileEntitySpecialRenderer {
+
+	/**
+	 * The TileEntityRenderer instance associated with this TileEntitySpecialRenderer
+	 */
 	protected TileEntityRenderer tileEntityRenderer;
 
 	public abstract void renderTileEntityAt(TileEntity var1, double var2, double var4, double var6, float var8);
 
-	protected void bindTexture(ResourceLocation var1) {
+	protected void bindTexture(ResourceLocation par1ResourceLocation) {
 		TextureManager var2 = this.tileEntityRenderer.renderEngine;
-		if(var2 != null) {
-			var2.bindTexture(var1);
+
+		if (var2 != null) {
+			var2.bindTexture(par1ResourceLocation);
 		}
-
-	}
-
-	public void setTileEntityRenderer(TileEntityRenderer var1) {
-		this.tileEntityRenderer = var1;
-	}
-
-	public void onWorldChange(World var1) {
-	}
+	}
+
+	/**
+	 * Associate a TileEntityRenderer with this TileEntitySpecialRenderer
+	 */
+	public void setTileEntityRenderer(TileEntityRenderer par1TileEntityRenderer) {
+		this.tileEntityRenderer = par1TileEntityRenderer;
+	}
+
+	/**
+	 * Called when the ingame world being rendered changes (e.g. on world -> nether travel) due to using one renderer per
+	 * tile entity type, rather than instance
+	 */
+	public void onWorldChange(World par1World) {}
 
 	public FontRenderer getFontRenderer() {
 		return this.tileEntityRenderer.getFontRenderer();
