--- net/minecraft/src/TileEntityMobSpawnerLogic.java
+++ net/minecraft/src/TileEntityMobSpawnerLogic.java
@@ -1,14 +1,16 @@
 package net.minecraft.src;
 
 class TileEntityMobSpawnerLogic extends MobSpawnerBaseLogic {
+
+	/** The mob spawner we deal with */
 	final TileEntityMobSpawner mobSpawnerEntity;
 
-	TileEntityMobSpawnerLogic(TileEntityMobSpawner var1) {
-		this.mobSpawnerEntity = var1;
+	TileEntityMobSpawnerLogic(TileEntityMobSpawner par1TileEntityMobSpawner) {
+		this.mobSpawnerEntity = par1TileEntityMobSpawner;
 	}
 
-	public void func_98267_a(int var1) {
-		this.mobSpawnerEntity.worldObj.addBlockEvent(this.mobSpawnerEntity.xCoord, this.mobSpawnerEntity.yCoord, this.mobSpawnerEntity.zCoord, Block.mobSpawner.blockID, var1, 0);
+	public void func_98267_a(int par1) {
+		this.mobSpawnerEntity.worldObj.addBlockEvent(this.mobSpawnerEntity.xCoord, this.mobSpawnerEntity.yCoord, this.mobSpawnerEntity.zCoord, Block.mobSpawner.blockID, par1, 0);
 	}
 
 	public World getSpawnerWorld() {
@@ -27,11 +29,11 @@
 		return this.mobSpawnerEntity.zCoord;
 	}
 
-	public void setRandomMinecart(WeightedRandomMinecart var1) {
-		super.setRandomMinecart(var1);
-		if(this.getSpawnerWorld() != null) {
+	public void setRandomMinecart(WeightedRandomMinecart par1WeightedRandomMinecart) {
+		super.setRandomMinecart(par1WeightedRandomMinecart);
+
+		if (this.getSpawnerWorld() != null) {
 			this.getSpawnerWorld().markBlockForUpdate(this.mobSpawnerEntity.xCoord, this.mobSpawnerEntity.yCoord, this.mobSpawnerEntity.zCoord);
 		}
-
 	}
 }
