--- net/minecraft/src/VillageDoorInfo.java
+++ net/minecraft/src/VillageDoorInfo.java
@@ -10,26 +10,33 @@
 	public boolean isDetachedFromVillageFlag;
 	private int doorOpeningRestrictionCounter;
 
-	public VillageDoorInfo(int var1, int var2, int var3, int var4, int var5, int var6) {
-		this.posX = var1;
-		this.posY = var2;
-		this.posZ = var3;
-		this.insideDirectionX = var4;
-		this.insideDirectionZ = var5;
-		this.lastActivityTimestamp = var6;
+	public VillageDoorInfo(int par1, int par2, int par3, int par4, int par5, int par6) {
+		this.posX = par1;
+		this.posY = par2;
+		this.posZ = par3;
+		this.insideDirectionX = par4;
+		this.insideDirectionZ = par5;
+		this.lastActivityTimestamp = par6;
 	}
 
-	public int getDistanceSquared(int var1, int var2, int var3) {
-		int var4 = var1 - this.posX;
-		int var5 = var2 - this.posY;
-		int var6 = var3 - this.posZ;
+	/**
+	 * Returns the squared distance between this door and the given coordinate.
+	 */
+	public int getDistanceSquared(int par1, int par2, int par3) {
+		int var4 = par1 - this.posX;
+		int var5 = par2 - this.posY;
+		int var6 = par3 - this.posZ;
 		return var4 * var4 + var5 * var5 + var6 * var6;
 	}
 
-	public int getInsideDistanceSquare(int var1, int var2, int var3) {
-		int var4 = var1 - this.posX - this.insideDirectionX;
-		int var5 = var2 - this.posY;
-		int var6 = var3 - this.posZ - this.insideDirectionZ;
+	/**
+	 * Get the square of the distance from a location 2 blocks away from the door considered 'inside' and the given
+	 * arguments
+	 */
+	public int getInsideDistanceSquare(int par1, int par2, int par3) {
+		int var4 = par1 - this.posX - this.insideDirectionX;
+		int var5 = par2 - this.posY;
+		int var6 = par3 - this.posZ - this.insideDirectionZ;
 		return var4 * var4 + var5 * var5 + var6 * var6;
 	}
 
@@ -45,9 +52,9 @@
 		return this.posZ + this.insideDirectionZ;
 	}
 
-	public boolean isInside(int var1, int var2) {
-		int var3 = var1 - this.posX;
-		int var4 = var2 - this.posZ;
+	public boolean isInside(int par1, int par2) {
+		int var3 = par1 - this.posX;
+		int var4 = par2 - this.posZ;
 		return var3 * this.insideDirectionX + var4 * this.insideDirectionZ >= 0;
 	}
 
