--- net/minecraft/src/WorldServerMulti.java
+++ net/minecraft/src/WorldServerMulti.java
@@ -3,13 +3,15 @@
 import net.minecraft.server.MinecraftServer;
 
 public class WorldServerMulti extends WorldServer {
-	public WorldServerMulti(MinecraftServer var1, ISaveHandler var2, String var3, int var4, WorldSettings var5, WorldServer var6, Profiler var7, ILogAgent var8) {
-		super(var1, var2, var3, var4, var5, var7, var8);
-		this.mapStorage = var6.mapStorage;
-		this.worldScoreboard = var6.getScoreboard();
-		this.worldInfo = new DerivedWorldInfo(var6.getWorldInfo());
+	public WorldServerMulti(MinecraftServer par1MinecraftServer, ISaveHandler par2ISaveHandler, String par3Str, int par4, WorldSettings par5WorldSettings, WorldServer par6WorldServer, Profiler par7Profiler, ILogAgent par8ILogAgent) {
+		super(par1MinecraftServer, par2ISaveHandler, par3Str, par4, par5WorldSettings, par7Profiler, par8ILogAgent);
+		this.mapStorage = par6WorldServer.mapStorage;
+		this.worldScoreboard = par6WorldServer.getScoreboard();
+		this.worldInfo = new DerivedWorldInfo(par6WorldServer.getWorldInfo());
 	}
 
-	protected void saveLevel() throws MinecraftException {
-	}
+	/**
+	 * Saves the chunks to disk.
+	 */
+	protected void saveLevel() throws MinecraftException {}
 }
